diff --git a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp b/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
index e58e22d..cd232ff 100644
--- a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
@@ -185,7 +185,9 @@ Hunk #2, a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp void CCallbackConsoleBase::CommonError(const FString &path, DWORD systemError, b
 
     *_se << endl << (isWarning ? kWarning : kError)
         << NError::MyFormatMessage(systemError)
-        << endl << fs2us(path) << endl << endl;
+        << endl;
+    _se->NormalizePrint_UString(fs2us(path));
+    *_se << endl << endl;
     _se->Flush();
   }
 }
