diff --git a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp b/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
index e58e22d..cd232ff 100644
--- a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
@@ -281,8 +281,8 @@ Hunk #3, a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp HRESULT CUpdateCallbackConsole::StartArchive(const wchar_t *name, bool updating)
   if (_so)
   {
     *_so << (updating ? kUpdatingArchiveMessage : kCreatingArchiveMessage);
-    if (name != 0)
-      *_so << name;
+    if (name)
+      _so->NormalizePrint_wstr(name);
     else
       *_so << k_StdOut_ArcName;
    *_so << endl << endl;
