diff --git a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp b/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
index e58e22d..cd232ff 100644
--- a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
@@ -475,7 +475,10 @@ Hunk #5, a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp HRESULT CCallbackConsoleBase::PrintProgress(const wchar_t *name, const char *com
 
     _tempU.Empty();
     if (name)
+    {
       _tempU = name;
+      _so->Normalize_UString(_tempU);
+    }
     _so->PrintUString(_tempU, _tempA);
     *_so << endl;
     if (NeedFlush)
