diff --git a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp b/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
index e58e22d..cd232ff 100644
--- a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp
@@ -568,7 +568,9 @@ Hunk #6, a/CPP/7zip/UI/Console/UpdateCallbackConsole.cpp HRESULT CUpdateCallbackConsole::ReportExtractResult(Int32 opRes, Int32 isEncrypt
 
       AString s;
       SetExtractErrorMessage(opRes, isEncrypted, s);
-      *_se << s << " : " << endl << name << endl << endl;
+      *_se << s << " : " << endl;
+      _se->NormalizePrint_wstr(name);
+      *_se << endl << endl;
       _se->Flush();
     }
     return S_OK;
