diff --git a/CPP/7zip/UI/Explorer/ContextMenu.cpp b/CPP/7zip/UI/Explorer/ContextMenu.cpp
index 21de12a..a99ca81 100644
--- a/CPP/7zip/UI/Explorer/ContextMenu.cpp
+++ b/CPP/7zip/UI/Explorer/ContextMenu.cpp
@@ -423,6 +423,13 @@ Hunk #2, a/CPP/7zip/UI/Explorer/ContextMenu.cpp STDMETHODIMP CZipContextMenu::QueryContextMenu(HMENU hMenu, UINT indexMenu,
 {
   // OutputDebugStringA("QueryContextMenu");
 
+  /*
+  for (UInt32 i = 0; i < _fileNames.Size(); i++)
+  {
+    OutputDebugStringW(_fileNames[i]);
+  }
+  */
+
   LoadLangOneTime();
   if (_fileNames.Size() == 0)
     return E_FAIL;
