diff --git a/CPP/7zip/UI/FileManager/FM.mak b/CPP/7zip/UI/FileManager/FM.mak
index 6a85ea2..654da62 100644
--- a/CPP/7zip/UI/FileManager/FM.mak
+++ b/CPP/7zip/UI/FileManager/FM.mak
@@ -46,6 +46,7 @@ Hunk #2, a/CPP/7zip/UI/FileManager/FM.mak FM_OBJS = \
   $O\AboutDialog.obj \
   $O\ComboDialog.obj \
   $O\CopyDialog.obj \
+  $O\EditDialog.obj \
   $O\EditPage.obj \
   $O\LangPage.obj \
   $O\ListViewDialog.obj \
