diff --git a/CPP/7zip/UI/FileManager/ListViewDialog.cpp b/CPP/7zip/UI/FileManager/ListViewDialog.cpp
index 80cdd7f..a42e790 100644
--- a/CPP/7zip/UI/FileManager/ListViewDialog.cpp
+++ b/CPP/7zip/UI/FileManager/ListViewDialog.cpp
@@ -11,6 +11,23 @@ Hunk #2, a/CPP/7zip/UI/FileManager/ListViewDialog.cpp
 
 using namespace NWindows;
 
+static const unsigned kOneStringMaxSize = 1024;
+
+
+static void ListView_GetSelected(NControl::CListView &listView, CUIntVector &vector)
+{
+  vector.Clear();
+  int index = -1;
+  for (;;)
+  {
+    index = listView.GetNextSelectedItem(index);
+    if (index < 0)
+      break;
+    vector.Add(index);
+  }
+}
+
+
 bool CListViewDialog::OnInit()
 {
   #ifdef LANG
