diff --git a/CPP/7zip/UI/FileManager/ListViewDialog.cpp b/CPP/7zip/UI/FileManager/ListViewDialog.cpp
index 80cdd7f..a42e790 100644
--- a/CPP/7zip/UI/FileManager/ListViewDialog.cpp
+++ b/CPP/7zip/UI/FileManager/ListViewDialog.cpp
@@ -18,28 +18,92 @@ Hunk #3, a/CPP/7zip/UI/FileManager/ListViewDialog.cpp bool CListViewDialog::OnInit()
   #endif
   _listView.Attach(GetItem(IDL_LISTVIEW));
 
+  if (NumColumns > 1)
+  {
+    LONG_PTR style = _listView.GetStyle();
+    style &= ~(LONG_PTR)LVS_NOCOLUMNHEADER;
+    _listView.SetStyle(style);
+  }
+
   CFmSettings st;
   st.Load();
+
+  DWORD exStyle = 0;
+  
   if (st.SingleClick)
-    _listView.SetExtendedListViewStyle(LVS_EX_ONECLICKACTIVATE | LVS_EX_TRACKSELECT);
+    exStyle |= LVS_EX_ONECLICKACTIVATE | LVS_EX_TRACKSELECT;
+
+  exStyle |= LVS_EX_FULLROWSELECT;
+  if (exStyle != 0)
+    _listView.SetExtendedListViewStyle(exStyle);
+
 
   SetText(Title);
 
+  const int kWidth = 400;
+  
   LVCOLUMN columnInfo;
   columnInfo.mask = LVCF_FMT | LVCF_WIDTH | LVCF_SUBITEM;
   columnInfo.fmt = LVCFMT_LEFT;
   columnInfo.iSubItem = 0;
-  columnInfo.cx = 200;
+  columnInfo.cx = kWidth;
+  columnInfo.pszText = NULL; // (TCHAR *)(const TCHAR *)""; // "Property"
 
-  _listView.InsertColumn(0, &columnInfo);
+  if (NumColumns > 1)
+  {
+    columnInfo.cx = 100;
+    /*
+    // Windows always uses LVCFMT_LEFT for first column.
+    // if we need LVCFMT_RIGHT, we can create dummy column and then remove it
 
+    // columnInfo.mask |= LVCF_TEXT;
+    _listView.InsertColumn(0, &columnInfo);
+  
+    columnInfo.iSubItem = 1;
+    columnInfo.fmt = LVCFMT_RIGHT;
+    _listView.InsertColumn(1, &columnInfo);
+    _listView.DeleteColumn(0);
+    */
+  }
+  // else
+    _listView.InsertColumn(0, &columnInfo);
+
+  if (NumColumns > 1)
+  {
+    // columnInfo.fmt = LVCFMT_LEFT;
+    columnInfo.cx = kWidth - columnInfo.cx;
+    columnInfo.iSubItem = 1;
+    // columnInfo.pszText = NULL; // (TCHAR *)(const TCHAR *)""; // "Value"
+    _listView.InsertColumn(1, &columnInfo);
+  }
+
+
+  UString s;
+  
   FOR_VECTOR (i, Strings)
+  {
     _listView.InsertItem(i, Strings[i]);
 
-  if (Strings.Size() > 0)
+    if (NumColumns > 1 && i < Values.Size())
+    {
+      s = Values[i];
+      if (s.Len() > kOneStringMaxSize)
+      {
+        s.DeleteFrom(kOneStringMaxSize);
+        s += " ...";
+      }
+      s.Replace(L"\r\n", L" ");
+      s.Replace(L"\n", L" ");
+      _listView.SetSubItem(i, 1, s);
+    }
+  }
+
+  if (SelectFirst && Strings.Size() > 0)
     _listView.SetItemState_FocusedSelected(0);
 
   _listView.SetColumnWidthAuto(0);
+  if (NumColumns > 1)
+    _listView.SetColumnWidthAuto(1);
   StringsWereChanged = false;
 
   NormalizeSize();
