diff --git a/CPP/7zip/UI/FileManager/ListViewDialog.cpp b/CPP/7zip/UI/FileManager/ListViewDialog.cpp
index 80cdd7f..a42e790 100644
--- a/CPP/7zip/UI/FileManager/ListViewDialog.cpp
+++ b/CPP/7zip/UI/FileManager/ListViewDialog.cpp
@@ -93,7 +93,7 @@ Hunk #6, a/CPP/7zip/UI/FileManager/ListViewDialog.cpp bool CListViewDialog::OnNotify(UINT /* controlID */, LPNMHDR header)
     case NM_RETURN: // probabably it's unused
       if (!g_LVN_ITEMACTIVATE_Support)
       {
-        OnOK();
+        OnEnter();
         return true;
       }
       break;
