diff --git a/CPP/7zip/UI/FileManager/PanelCopy.cpp b/CPP/7zip/UI/FileManager/PanelCopy.cpp
index d07def1..c3416cc 100644
--- a/CPP/7zip/UI/FileManager/PanelCopy.cpp
+++ b/CPP/7zip/UI/FileManager/PanelCopy.cpp
@@ -69,11 +69,15 @@ Hunk #3, a/CPP/7zip/UI/FileManager/PanelCopy.cpp HRESULT CPanelCopyThread::ProcessVirt()
       BoolToInt(options->replaceAltStreamChars),
       options->folder, ExtractCallback);
 
-  if (Result == S_OK && !ExtractCallbackSpec->ThereAreMessageErrors &&
-      (!options->hashMethods.IsEmpty() || options->testMode))
+  if (Result == S_OK && !ExtractCallbackSpec->ThereAreMessageErrors)
   {
-    CProgressMessageBoxPair &pair = GetMessagePair(false); // GetMessagePair(ExtractCallbackSpec->Hash.NumErrors != 0);
-    AddHashBundleRes(pair.Message, Hash, FirstFilePath);
+    if (!options->hashMethods.IsEmpty())
+      NeedShowRes = true;
+    else if (options->testMode)
+    {
+      CProgressMessageBoxPair &pair = GetMessagePair(false); // GetMessagePair(ExtractCallbackSpec->Hash.NumErrors != 0);
+      AddHashBundleRes(pair.Message, Hash);
+    }
   }
 
   return Result;
