diff --git a/CPP/7zip/UI/FileManager/PanelCopy.cpp b/CPP/7zip/UI/FileManager/PanelCopy.cpp
index d07def1..c3416cc 100644
--- a/CPP/7zip/UI/FileManager/PanelCopy.cpp
+++ b/CPP/7zip/UI/FileManager/PanelCopy.cpp
@@ -208,6 +208,9 @@ Hunk #10, a/CPP/7zip/UI/FileManager/PanelCopy.cpp HRESULT CPanel::CopyTo(CCopyToOptions &options, const CRecordVector<UInt32> &ind
     usePassword = extracter.ExtractCallbackSpec->PasswordIsDefined;
     password = extracter.ExtractCallbackSpec->Password;
   }
+
+  extracter.ShowFinalResults(_window);
+
   }
   
   RefreshTitleAlways();
