diff --git a/CPP/7zip/UI/FileManager/PanelCopy.cpp b/CPP/7zip/UI/FileManager/PanelCopy.cpp
index d07def1..c3416cc 100644
--- a/CPP/7zip/UI/FileManager/PanelCopy.cpp
+++ b/CPP/7zip/UI/FileManager/PanelCopy.cpp
@@ -296,8 +296,8 @@ Hunk #11, a/CPP/7zip/UI/FileManager/PanelCopy.cpp HRESULT CPanel::CopyFrom(bool moveMode, const UString &folderPrefix, const UStri
   NWindows::CThread thread;
   RINOK(thread.Create(CThreadUpdate::MyThreadFunction, &updater));
   updater.ProgressDialog.Create(title, thread, GetParent());
-  
-  if (messages != 0)
+
+  if (messages)
     *messages = updater.ProgressDialog.Sync.Messages;
 
   res = updater.Result;
