diff --git a/CPP/7zip/UI/FileManager/PanelCrc.cpp b/CPP/7zip/UI/FileManager/PanelCrc.cpp
index 227c453..1cfbfe5 100644
--- a/CPP/7zip/UI/FileManager/PanelCrc.cpp
+++ b/CPP/7zip/UI/FileManager/PanelCrc.cpp
@@ -233,7 +233,6 @@ Hunk #3, a/CPP/7zip/UI/FileManager/PanelCrc.cpp HRESULT CThreadCrc::ProcessVirt()
   Enumerator.Init();
 
   FString tempPath;
-  FString firstFilePath;
   bool isFirstFile = true;
   UInt64 errorsFilesSize = 0;
 
