diff --git a/CPP/7zip/UI/FileManager/PanelCrc.cpp b/CPP/7zip/UI/FileManager/PanelCrc.cpp
index 227c453..1cfbfe5 100644
--- a/CPP/7zip/UI/FileManager/PanelCrc.cpp
+++ b/CPP/7zip/UI/FileManager/PanelCrc.cpp
@@ -264,7 +264,7 @@ Hunk #4, a/CPP/7zip/UI/FileManager/PanelCrc.cpp HRESULT CThreadCrc::ProcessVirt()
       }
       if (isFirstFile)
       {
-        firstFilePath = path;
+        Hash.FirstFileName = path;
         isFirstFile = false;
       }
       sync.Set_FilePath(fs2us(path));
