diff --git a/CPP/7zip/UI/FileManager/PanelCrc.cpp b/CPP/7zip/UI/FileManager/PanelCrc.cpp
index 227c453..1cfbfe5 100644
--- a/CPP/7zip/UI/FileManager/PanelCrc.cpp
+++ b/CPP/7zip/UI/FileManager/PanelCrc.cpp
@@ -338,14 +338,19 @@ Hunk #6, a/CPP/7zip/UI/FileManager/PanelCrc.cpp HRESULT CApp::CalculateCrc2(const UString &methodName)
 
   {
     CThreadCrc t;
+
     {
       UStringVector methods;
       methods.Add(methodName);
       RINOK(t.Hash.SetMethods(EXTERNAL_CODECS_VARS_G methods));
     }
+    
     FOR_VECTOR (i, indices)
       t.Enumerator.FilePaths.Add(us2fs(srcPanel.GetItemRelPath(indices[i])));
 
+    if (t.Enumerator.FilePaths.Size() == 1)
+      t.Hash.MainName = t.Enumerator.FilePaths[0];
+
     UString basePrefix = srcPanel.GetFsPath();
     UString basePrefix2 = basePrefix;
     if (basePrefix2.Back() == ':')
