diff --git a/CPP/7zip/UI/FileManager/PanelDrag.cpp b/CPP/7zip/UI/FileManager/PanelDrag.cpp
index b6fca59..6a781e2 100644
--- a/CPP/7zip/UI/FileManager/PanelDrag.cpp
+++ b/CPP/7zip/UI/FileManager/PanelDrag.cpp
@@ -662,6 +662,21 @@ Hunk #13, a/CPP/7zip/UI/FileManager/PanelDrag.cpp bool CDropTarget::IsItSameDrive() const
   return true;
 }
 
+
+/*
+  There are 2 different actions, when we drag to 7-Zip:
+  1) Drag from any external program except of Explorer to "7-Zip" FS folder.
+     We want to create new archive for that operation.
+  2) all another operation work as usual file COPY/MOVE
+    - Drag from "7-Zip" FS to "7-Zip" FS.
+        COPY/MOVE are supported.
+    - Drag to open archive in 7-Zip.
+        We want to update archive.
+        We replace COPY to MOVE.
+    - Drag from "7-Zip" archive to "7-Zip" FS.
+        We replace COPY to MOVE.
+*/
+
 DWORD CDropTarget::GetEffect(DWORD keyState, POINTL /* pt */, DWORD allowedEffect)
 {
   if (!m_DropIsAllowed || !m_PanelDropIsAllowed)
