diff --git a/CPP/7zip/UI/FileManager/PanelItemOpen.cpp b/CPP/7zip/UI/FileManager/PanelItemOpen.cpp
index f7c05a3..89f49bf 100644
--- a/CPP/7zip/UI/FileManager/PanelItemOpen.cpp
+++ b/CPP/7zip/UI/FileManager/PanelItemOpen.cpp
@@ -674,7 +674,7 @@ Hunk #1, a/CPP/7zip/UI/FileManager/PanelItemOpen.cpp static const char * const kStartExtensions =
   #endif
   " exe bat ps1 com"
   " chm"
-  " msi doc xls ppt pps wps wpt wks xlr wdb vsd pub"
+  " msi doc dot xls ppt pps wps wpt wks xlr wdb vsd pub"
 
   " docx docm dotx dotm xlsx xlsm xltx xltm xlsb xps"
   " xlam pptx pptm potx potm ppam ppsx ppsm xsn"
