diff --git a/CPP/7zip/UI/FileManager/PanelMenu.cpp b/CPP/7zip/UI/FileManager/PanelMenu.cpp
index 945a41f..11c6d7c 100644
--- a/CPP/7zip/UI/FileManager/PanelMenu.cpp
+++ b/CPP/7zip/UI/FileManager/PanelMenu.cpp
@@ -352,10 +352,15 @@ Hunk #10, a/CPP/7zip/UI/FileManager/PanelMenu.cpp void CPanel::Properties()
         }
       }
     }
-    ::MessageBoxW(*(this), message, LangString(IDS_PROPERTIES), MB_OK);
+
+    message.Title = LangString(IDS_PROPERTIES);
+    message.NumColumns = 2;
+    message.Create(GetParent());
   }
 }
 
+
+
 void CPanel::EditCut()
 {
   // InvokeSystemCommand("cut");
