diff --git a/CPP/7zip/UI/FileManager/PanelSelect.cpp b/CPP/7zip/UI/FileManager/PanelSelect.cpp
index 268b64a..8cfb23f 100644
--- a/CPP/7zip/UI/FileManager/PanelSelect.cpp
+++ b/CPP/7zip/UI/FileManager/PanelSelect.cpp
@@ -218,6 +218,8 @@ Hunk #1, a/CPP/7zip/UI/FileManager/PanelSelect.cpp void CPanel::InvertSelection()
     FOR_VECTOR (i, _selectedStatusVector)
       if (_selectedStatusVector[i])
         numSelected++;
+    // 17.02: fixed : now we invert item even, if single item is selected
+    /*
     if (numSelected == 1)
     {
       int focused = _listView.GetFocusedItem();
