diff --git a/CPP/7zip/UI/FileManager/PanelSelect.cpp b/CPP/7zip/UI/FileManager/PanelSelect.cpp
index 268b64a..8cfb23f 100644
--- a/CPP/7zip/UI/FileManager/PanelSelect.cpp
+++ b/CPP/7zip/UI/FileManager/PanelSelect.cpp
@@ -229,6 +229,7 @@ Hunk #2, a/CPP/7zip/UI/FileManager/PanelSelect.cpp void CPanel::InvertSelection()
             _selectedStatusVector[realIndex] = false;
       }
     }
+    */
   }
   FOR_VECTOR (i, _selectedStatusVector)
     _selectedStatusVector[i] = !_selectedStatusVector[i];
