diff --git a/CPP/7zip/UI/FileManager/PanelSplitFile.cpp b/CPP/7zip/UI/FileManager/PanelSplitFile.cpp
index 3f7111a..7c63332 100644
--- a/CPP/7zip/UI/FileManager/PanelSplitFile.cpp
+++ b/CPP/7zip/UI/FileManager/PanelSplitFile.cpp
@@ -235,7 +235,7 @@ Hunk #4, a/CPP/7zip/UI/FileManager/PanelSplitFile.cpp void CApp::Split()
   CThreadSplit spliter;
   spliter.NumVolumes = numVolumes;
 
-  CProgressDialog &progressDialog = spliter.ProgressDialog;
+  CProgressDialog &progressDialog = spliter;
 
   UString progressWindowTitle ("7-Zip"); // LangString(IDS_APP_TITLE, 0x03000000);
   UString title = LangString(IDS_SPLITTING);
