diff --git a/CPP/7zip/UI/FileManager/PanelSplitFile.cpp b/CPP/7zip/UI/FileManager/PanelSplitFile.cpp
index 3f7111a..7c63332 100644
--- a/CPP/7zip/UI/FileManager/PanelSplitFile.cpp
+++ b/CPP/7zip/UI/FileManager/PanelSplitFile.cpp
@@ -291,7 +291,7 @@ Hunk #5, a/CPP/7zip/UI/FileManager/PanelSplitFile.cpp HRESULT CThreadCombine::ProcessVirt()
     return res;
   }
   
-  CProgressSync &sync = ProgressDialog.Sync;
+  CProgressSync &sync = Sync;
   sync.Set_NumBytesTotal(TotalSize);
   
   CMyBuffer bufferObject;
