diff --git a/C/Lzma2Enc.c b/C/Lzma2Enc.c
index e970944..5c1ad49 100644
--- a/C/Lzma2Enc.c
+++ b/C/Lzma2Enc.c
@@ -666,7 +666,7 @@ Hunk #5, a/C/Lzma2Enc.c static SRes Lzma2Enc_MtCallback_Code(void *pp, unsigned coderIndex, unsigned out
 
   if (!dest)
   {
-    dest = ISzAlloc_Alloc(me->alloc, me->outBufSize);
+    dest = (Byte *)ISzAlloc_Alloc(me->alloc, me->outBufSize);
     if (!dest)
       return SZ_ERROR_MEM;
     me->outBufs[outBufIndex] = dest;
