diff --git a/CPP/7zip/UI/FileManager/ProgressDialog2.cpp b/CPP/7zip/UI/FileManager/ProgressDialog2.cpp
index 762f249..28e3eae 100644
--- a/CPP/7zip/UI/FileManager/ProgressDialog2.cpp
+++ b/CPP/7zip/UI/FileManager/ProgressDialog2.cpp
@@ -974,7 +974,9 @@ Hunk #2, a/CPP/7zip/UI/FileManager/ProgressDialog2.cpp bool CProgressDialog::OnExternalCloseMessage()
   ::SendMessage(GetItem(IDCANCEL), BM_SETSTYLE, BS_DEFPUSHBUTTON, MAKELPARAM(TRUE, 0));
   HideItem(IDB_PROGRESS_BACKGROUND);
   HideItem(IDB_PAUSE);
-  
+
+  ProcessWasFinished_GuiVirt();
+
   bool thereAreMessages;
   CProgressFinalMessage fm;
   {
