diff --git a/CPP/7zip/UI/FileManager/ProgressDialog2.cpp b/CPP/7zip/UI/FileManager/ProgressDialog2.cpp
index 762f249..28e3eae 100644
--- a/CPP/7zip/UI/FileManager/ProgressDialog2.cpp
+++ b/CPP/7zip/UI/FileManager/ProgressDialog2.cpp
@@ -982,6 +982,7 @@ Hunk #3, a/CPP/7zip/UI/FileManager/ProgressDialog2.cpp bool CProgressDialog::OnExternalCloseMessage()
     thereAreMessages = !Sync.Messages.IsEmpty();
     fm = Sync.FinalMessage;
   }
+
   if (!fm.ErrorMessage.Message.IsEmpty())
   {
     MessagesDisplayed = true;
