diff --git a/CPP/7zip/UI/FileManager/ProgressDialog2.h b/CPP/7zip/UI/FileManager/ProgressDialog2.h
index 03f836d..6c4213a 100644
--- a/CPP/7zip/UI/FileManager/ProgressDialog2.h
+++ b/CPP/7zip/UI/FileManager/ProgressDialog2.h
@@ -261,7 +261,18 @@ Hunk #1, a/CPP/7zip/UI/FileManager/ProgressDialog2.h class CProgressDialog: public NWindows::NControl::CModalDialog
 
   INT_PTR Create(const UString &title, NWindows::CThread &thread, HWND wndParent = 0);
 
+
+  /* how it works:
+     1) the working thread calls ProcessWasFinished()
+        that sends kCloseMessage message to CProgressDialog (GUI) thread
+     2) CProgressDialog (GUI) thread receives kCloseMessage message and
+        calls ProcessWasFinished_GuiVirt();
+        So we can implement ProcessWasFinished_GuiVirt() and show special
+        results window in GUI thread with CProgressDialog as parent window
+  */
+
   void ProcessWasFinished();
+  virtual void ProcessWasFinished_GuiVirt() {}
 };
 
 
