diff --git a/C/Lzma2Enc.c b/C/Lzma2Enc.c
index e970944..5c1ad49 100644
--- a/C/Lzma2Enc.c
+++ b/C/Lzma2Enc.c
@@ -698,10 +698,10 @@ Hunk #7, a/C/Lzma2Enc.c static SRes Lzma2Enc_MtCallback_Write(void *pp, unsigned outBufIndex)
   if (me->outStream)
     return ISeqOutStream_Write(me->outStream, data, size) == size ? SZ_OK : SZ_ERROR_WRITE;
   
-  if (size > me->outBufSize)
+  if (size > me->outBuf_Rem)
     return SZ_ERROR_OUTPUT_EOF;
   memcpy(me->outBuf, data, size);
-  me->outBufSize -= size;
+  me->outBuf_Rem -= size;
   me->outBuf += size;
   return SZ_OK;
 }
