diff --git a/CPP/7zip/UI/GUI/ExtractGUI.cpp b/CPP/7zip/UI/GUI/ExtractGUI.cpp
index 5071d1f..1e37efb 100644
--- a/CPP/7zip/UI/GUI/ExtractGUI.cpp
+++ b/CPP/7zip/UI/GUI/ExtractGUI.cpp
@@ -109,16 +109,23 @@ Hunk #3, a/CPP/7zip/UI/GUI/ExtractGUI.cpp HRESULT CThreadExtracting::ProcessVirt()
         HashBundle,
       #endif
       FinalMessage.ErrorMessage.Message, Stat);
+  
   #ifndef _SFX
-  if (res == S_OK && Options->TestMode && ExtractCallbackSpec->IsOK())
+  if (res == S_OK && ExtractCallbackSpec->IsOK())
   {
-    UString s;
-    
-    AddValuePair(s, IDS_ARCHIVES_COLON, Stat.NumArchives, false);
-    AddSizePair(s, IDS_PROP_PACKED_SIZE, Stat.PackSize);
-
-    if (!HashBundle)
+    if (HashBundle)
     {
+      AddValuePair(Pairs, IDS_ARCHIVES_COLON, Stat.NumArchives);
+      AddSizeValuePair(Pairs, IDS_PROP_PACKED_SIZE, Stat.PackSize);
+      AddHashBundleRes(Pairs, *HashBundle);
+    }
+    else if (Options->TestMode)
+    {
+      UString s;
+    
+      AddValuePair(s, IDS_ARCHIVES_COLON, Stat.NumArchives, false);
+      AddSizePair(s, IDS_PROP_PACKED_SIZE, Stat.PackSize);
+
       if (Stat.NumFolders != 0)
         AddValuePair(s, IDS_PROP_FOLDERS, Stat.NumFolders);
       AddValuePair(s, IDS_PROP_FILES, Stat.NumFiles);
