diff --git a/CPP/7zip/UI/GUI/GUI.cpp b/CPP/7zip/UI/GUI/GUI.cpp
index 6bbfb74..2674ec9 100644
--- a/CPP/7zip/UI/GUI/GUI.cpp
+++ b/CPP/7zip/UI/GUI/GUI.cpp
@@ -397,13 +397,17 @@ Hunk #5, a/CPP/7zip/UI/GUI/GUI.cpp int APIENTRY WinMain(HINSTANCE  hInstance, HINSTANCE /* hPrevInstance */,
   // setlocale(LC_COLLATE, ".ACP");
   try
   {
+    #ifdef _WIN32
+    My_SetDefaultDllDirectories();
+    #endif
+
     return Main2();
   }
   catch(const CNewException &)
   {
     return ShowMemErrorMessage();
   }
-  catch(const CArcCmdLineException &e)
+  catch(const CMessagePathException &e)
   {
     ErrorMessage(e);
     return NExitCode::kUserError;
