diff --git a/C/Lzma2Enc.c b/C/Lzma2Enc.c
index e970944..5c1ad49 100644
--- a/C/Lzma2Enc.c
+++ b/C/Lzma2Enc.c
@@ -720,10 +720,10 @@ Hunk #8, a/C/Lzma2Enc.c SRes Lzma2Enc_Encode2(CLzma2EncHandle pp,
   CLzma2Enc *p = (CLzma2Enc *)pp;
 
   if (inStream && inData)
-    return E_INVALIDARG;
+    return SZ_ERROR_PARAM;
 
   if (outStream && outBuf)
-    return E_INVALIDARG;
+    return SZ_ERROR_PARAM;
 
   {
     unsigned i;
