diff --git a/CPP/7zip/UI/GUI/GUI.dsp b/CPP/7zip/UI/GUI/GUI.dsp
index c022088..b70e284 100644
--- a/CPP/7zip/UI/GUI/GUI.dsp
+++ b/CPP/7zip/UI/GUI/GUI.dsp
@@ -1051,6 +1051,14 @@ Hunk #5, a/CPP/7zip/UI/GUI/GUI.dsp SOURCE=..\..\..\Windows\FileSystem.h
 # End Source File
 # Begin Source File
 
+SOURCE=..\..\..\Windows\MemoryGlobal.cpp
+# End Source File
+# Begin Source File
+
+SOURCE=..\..\..\Windows\MemoryGlobal.h
+# End Source File
+# Begin Source File
+
 SOURCE=..\..\..\Windows\MemoryLock.cpp
 # End Source File
 # Begin Source File
