diff --git a/CPP/7zip/UI/GUI/HashGUI.cpp b/CPP/7zip/UI/GUI/HashGUI.cpp
index cc93f07..37cd65e 100644
--- a/CPP/7zip/UI/GUI/HashGUI.cpp
+++ b/CPP/7zip/UI/GUI/HashGUI.cpp
@@ -19,13 +19,19 @@ Hunk #2, a/CPP/7zip/UI/GUI/HashGUI.cpp
 
 using namespace NWindows;
 
+
+
 class CHashCallbackGUI: public CProgressThreadVirt, public IHashCallbackUI
 {
   UInt64 NumFiles;
   bool _curIsFolder;
   UString FirstFileName;
+  // UString MainPath;
+
+  CPropNameValPairs PropNameValPairs;
 
   HRESULT ProcessVirt();
+  virtual void ProcessWasFinished_GuiVirt();
 
 public:
   const NWildcard::CCensor *censor;
