diff --git a/CPP/7zip/UI/GUI/HashGUI.cpp b/CPP/7zip/UI/GUI/HashGUI.cpp
index cc93f07..37cd65e 100644
--- a/CPP/7zip/UI/GUI/HashGUI.cpp
+++ b/CPP/7zip/UI/GUI/HashGUI.cpp
@@ -98,26 +98,26 @@ Hunk #4, a/CPP/7zip/UI/GUI/HashGUI.cpp HRESULT CHashCallbackGUI::FinishScanning(const CDirItemsStat &st)
 
 HRESULT CHashCallbackGUI::CheckBreak()
 {
-  return ProgressDialog.Sync.CheckStop();
+  return Sync.CheckStop();
 }
 
 HRESULT CHashCallbackGUI::SetNumFiles(UInt64 numFiles)
 {
-  CProgressSync &sync = ProgressDialog.Sync;
+  CProgressSync &sync = Sync;
   sync.Set_NumFilesTotal(numFiles);
   return CheckBreak();
 }
 
 HRESULT CHashCallbackGUI::SetTotal(UInt64 size)
 {
-  CProgressSync &sync = ProgressDialog.Sync;
+  CProgressSync &sync = Sync;
   sync.Set_NumBytesTotal(size);
   return CheckBreak();
 }
 
 HRESULT CHashCallbackGUI::SetCompleted(const UInt64 *completed)
 {
-  return ProgressDialog.Sync.Set_NumBytesCur(completed);
+  return Sync.Set_NumBytesCur(completed);
 }
 
 HRESULT CHashCallbackGUI::BeforeFirstFile(const CHashBundle & /* hb */)
