diff --git a/CPP/Build.mak b/CPP/Build.mak
index 22b8e16..f53286b 100644
--- a/CPP/Build.mak
+++ b/CPP/Build.mak
@@ -29,16 +29,16 @@ Hunk #2, a/CPP/Build.mak RFLAGS = $(RFLAGS) -dUNDER_CE
 LFLAGS = $(LFLAGS) /ENTRY:mainACRTStartup
 !ENDIF
 !ELSE
-!IFNDEF NEW_COMPILER
+!IFDEF OLD_COMPILER
 LFLAGS = $(LFLAGS) -OPT:NOWIN98
 !ENDIF
-# !IF "$(CPU)" != "ARM"
+!IF "$(PLATFORM)" != "arm" && "$(PLATFORM)" != "arm64"
 CFLAGS = $(CFLAGS) -Gr
-# !ENDIF
+!ENDIF
 LIBS = $(LIBS) user32.lib advapi32.lib shell32.lib
 !ENDIF
 
-!IF "$(CPU)" == "ARM"
+!IF "$(PLATFORM)" == "arm"
 COMPL_ASM = $(MY_ML) $** $O/$(*B).obj
 !ELSE
 COMPL_ASM = $(MY_ML) -c -Fo$O/ $**
