diff --git a/CPP/Build.mak b/CPP/Build.mak
index 22b8e16..f53286b 100644
--- a/CPP/Build.mak
+++ b/CPP/Build.mak
@@ -46,19 +46,19 @@ Hunk #3, a/CPP/Build.mak COMPL_ASM = $(MY_ML) -c -Fo$O/ $**
 
 CFLAGS = $(CFLAGS) -nologo -c -Fo$O/ -W4 -WX -EHsc -Gy -GR- -GF
 
-!IFDEF MY_STATIC_LINK
+!IFDEF MY_DYNAMIC_LINK
+CFLAGS = $(CFLAGS) -MD
+!ELSE
 !IFNDEF MY_SINGLE_THREAD
 CFLAGS = $(CFLAGS) -MT
 !ENDIF
-!ELSE
-CFLAGS = $(CFLAGS) -MD
 !ENDIF
 
-!IFDEF NEW_COMPILER
+!IFNDEF OLD_COMPILER
 CFLAGS = $(CFLAGS) -GS- -Zc:forScope -Zc:wchar_t
 !IFNDEF UNDER_CE
 CFLAGS = $(CFLAGS) -MP2
-!IFNDEF CPU
+!IFNDEF PLATFORM
 # CFLAGS = $(CFLAGS) -arch:IA32
 !ENDIF
 !ENDIF
