diff --git a/CPP/Build.mak b/CPP/Build.mak
index 22b8e16..f53286b 100644
--- a/CPP/Build.mak
+++ b/CPP/Build.mak
@@ -66,7 +66,17 @@ Hunk #4, a/CPP/Build.mak CFLAGS = $(CFLAGS) -MP2
 CFLAGS = $(CFLAGS)
 !ENDIF
 
-!IF "$(CPU)" == "AMD64"
+!IFDEF MY_CONSOLE
+CFLAGS = $(CFLAGS) -D_CONSOLE
+!ENDIF
+
+!IFNDEF UNDER_CE
+!IF "$(PLATFORM)" == "arm"
+CFLAGS = $(CFLAGS) -D_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE
+!ENDIF
+!ENDIF
+
+!IF "$(PLATFORM)" == "x64"
 CFLAGS_O1 = $(CFLAGS) -O1
 !ELSE
 CFLAGS_O1 = $(CFLAGS) -O1
