diff --git a/CPP/Common/ListFileUtils.cpp b/CPP/Common/ListFileUtils.cpp
index 9112003..edd37eb 100644
--- a/CPP/Common/ListFileUtils.cpp
+++ b/CPP/Common/ListFileUtils.cpp
@@ -25,14 +25,21 @@ Hunk #1, a/CPP/Common/ListFileUtils.cpp static void AddName(UStringVector &strings, UString &s)
     strings.Add(s);
 }
 
-bool ReadNamesFromListFile(CFSTR fileName, UStringVector &strings, UINT codePage)
+bool ReadNamesFromListFile2(CFSTR fileName, UStringVector &strings, UINT codePage, DWORD &lastError)
 {
+  lastError = 0;
   NWindows::NFile::NIO::CInFile file;
   if (!file.Open(fileName))
+  {
+    lastError = ::GetLastError();
     return false;
+  }
   UInt64 fileSize;
   if (!file.GetLength(fileSize))
+  {
+    lastError = ::GetLastError();
     return false;
+  }
   if (fileSize >= ((UInt32)1 << 31) - 32)
     return false;
   UString u;
