diff --git a/C/Lzma86Enc.c b/C/Lzma86Enc.c
index a15b58d..2617bab 100644
--- a/C/Lzma86Enc.c
+++ b/C/Lzma86Enc.c
@@ -18,7 +18,7 @@ Hunk #2, a/C/Lzma86Enc.c int Lzma86_Encode(Byte *dest, size_t *destLen, const Byte *src, size_t srcLen,
 {
   size_t outSize2 = *destLen;
   Byte *filteredStream;
-  Bool useFilter;
+  BoolInt useFilter;
   int mainResult = SZ_ERROR_OUTPUT_EOF;
   CLzmaEncProps props;
   LzmaEncProps_Init(&props);
