diff --git a/CPP/Common/ListFileUtils.h b/CPP/Common/ListFileUtils.h
index e8d833f..a91e4b1 100644
--- a/CPP/Common/ListFileUtils.h
+++ b/CPP/Common/ListFileUtils.h
@@ -9,6 +9,10 @@ Hunk #1, a/CPP/Common/ListFileUtils.h
 #define MY__CP_UTF16   1200
 #define MY__CP_UTF16BE 1201
 
-bool ReadNamesFromListFile(CFSTR fileName, UStringVector &strings, UINT codePage = CP_OEMCP);
+// bool ReadNamesFromListFile(CFSTR fileName, UStringVector &strings, UINT codePage = CP_OEMCP);
+
+ // = CP_OEMCP
+bool ReadNamesFromListFile2(CFSTR fileName, UStringVector &strings, UINT codePage,
+    DWORD &lastError);
 
 #endif
