diff --git a/CPP/Common/MyBuffer2.h b/CPP/Common/MyBuffer2.h
index ba98d13..d61a72e 100644
--- a/CPP/Common/MyBuffer2.h
+++ b/CPP/Common/MyBuffer2.h
@@ -15,7 +15,7 @@ Hunk #2, a/CPP/Common/MyBuffer2.h class CMidBuffer
   CLASS_NO_COPY(CMidBuffer)
 
 public:
-  CMidBuffer(): _data(NULL), _size(0) {};
+  CMidBuffer(): _data(NULL), _size(0) {}
   ~CMidBuffer() { ::MidFree(_data); }
 
   void Free() { ::MidFree(_data); _data = NULL; _size = 0; }
