diff --git a/CPP/Common/MyString.h b/CPP/Common/MyString.h
index 1c7357e..40de52c 100644
--- a/CPP/Common/MyString.h
+++ b/CPP/Common/MyString.h
@@ -571,7 +571,7 @@ Hunk #3, a/CPP/Common/MyString.h class UString
   }
 
   UString &operator=(wchar_t c);
-  UString &operator=(char c) { return (*this)=((wchar_t)c); }
+  UString &operator=(char c) { return (*this)=((wchar_t)(unsigned char)c); }
   UString &operator=(const wchar_t *s);
   UString &operator=(const UString &s);
   void SetFrom(const wchar_t *s, unsigned len); // no check
