diff --git a/CPP/Common/StdOutStream.h b/CPP/Common/StdOutStream.h
index 1837bfb..2e637e8 100644
--- a/CPP/Common/StdOutStream.h
+++ b/CPP/Common/StdOutStream.h
@@ -13,7 +13,9 @@ Hunk #1, a/CPP/Common/StdOutStream.h class CStdOutStream
   FILE *_stream;
   bool _streamIsOpen;
 public:
-  CStdOutStream(): _stream(0), _streamIsOpen(false) {};
+  bool IsTerminalMode;
+
+  CStdOutStream(): _stream(0), _streamIsOpen(false), IsTerminalMode(false) {};
   CStdOutStream(FILE *stream): _stream(stream), _streamIsOpen(false) {};
   ~CStdOutStream() { Close(); }
 
