diff --git a/CPP/Common/StdOutStream.h b/CPP/Common/StdOutStream.h
index 1837bfb..2e637e8 100644
--- a/CPP/Common/StdOutStream.h
+++ b/CPP/Common/StdOutStream.h
@@ -50,6 +50,13 @@ Hunk #2, a/CPP/Common/StdOutStream.h class CStdOutStream
 
   CStdOutStream & operator<<(const wchar_t *s);
   void PrintUString(const UString &s, AString &temp);
+
+  void Normalize_UString__LF_Allowed(UString &s);
+  void Normalize_UString(UString &s);
+
+  void NormalizePrint_UString(const UString &s, UString &tempU, AString &tempA);
+  void NormalizePrint_UString(const UString &s);
+  void NormalizePrint_wstr(const wchar_t *s);
 };
 
 CStdOutStream & endl(CStdOutStream & outStream) throw();
