diff --git a/CPP/Windows/Control/Dialog.cpp b/CPP/Windows/Control/Dialog.cpp
index 9df3ef5..020694a 100644
--- a/CPP/Windows/Control/Dialog.cpp
+++ b/CPP/Windows/Control/Dialog.cpp
@@ -72,6 +72,7 @@ Hunk #1, a/CPP/Windows/Control/Dialog.cpp bool CDialog::OnButtonClicked(int buttonID, HWND /* buttonHWND */)
   {
     case IDOK: OnOK(); break;
     case IDCANCEL: OnCancel(); break;
+    case IDCLOSE: OnClose(); break;
     case IDHELP: OnHelp(); break;
     default: return false;
   }
