diff --git a/CPP/Windows/FileDir.cpp b/CPP/Windows/FileDir.cpp
index 72e0b7f..62b11c1 100644
--- a/CPP/Windows/FileDir.cpp
+++ b/CPP/Windows/FileDir.cpp
@@ -667,12 +667,27 @@ Hunk #1, a/CPP/Windows/FileDir.cpp bool CTempFile::Remove()
 
 bool CTempFile::MoveTo(CFSTR name, bool deleteDestBefore)
 {
+  // DWORD attrib = 0;
   if (deleteDestBefore)
+  {
     if (NFind::DoesFileExist(name))
+    {
+      // attrib = NFind::GetFileAttrib(name);
       if (!DeleteFileAlways(name))
         return false;
+    }
+  }
   DisableDeleting();
   return MyMoveFile(_path, name);
+  
+  /*
+  if (attrib != INVALID_FILE_ATTRIBUTES && (attrib & FILE_ATTRIBUTE_READONLY))
+  {
+    DWORD attrib2 = NFind::GetFileAttrib(name);
+    if (attrib2 != INVALID_FILE_ATTRIBUTES)
+      SetFileAttrib(name, attrib2 | FILE_ATTRIBUTE_READONLY);
+  }
+  */
 }
 
 bool CTempDir::Create(CFSTR prefix)
