diff --git a/CPP/Windows/FileLink.cpp b/CPP/Windows/FileLink.cpp
index 9e706ea..3e2f643 100644
--- a/CPP/Windows/FileLink.cpp
+++ b/CPP/Windows/FileLink.cpp
@@ -194,8 +194,9 @@ Hunk #1, a/CPP/Windows/FileLink.cpp static void GetString(const Byte *p, unsigned len, UString &res)
   res.ReleaseBuf_SetLen(i);
 }
 
-bool CReparseAttr::Parse(const Byte *p, size_t size)
+bool CReparseAttr::Parse(const Byte *p, size_t size, DWORD &errorCode)
 {
+  errorCode = ERROR_INVALID_REPARSE_DATA;
   if (size < 8)
     return false;
   Tag = Get32(p);
