diff --git a/DOC/7zip.nsi b/DOC/7zip.nsi
index 5ba331d..5a806ae 100644
--- a/DOC/7zip.nsi
+++ b/DOC/7zip.nsi
@@ -1,9 +1,9 @@ Hunk #1, a/DOC/7zip.nsi
 ;--------------------------------
 ;Defines
 
-!define VERSION_MAJOR 17
-!define VERSION_MINOR 01
-!define VERSION_POSTFIX_FULL " beta"
+!define VERSION_MAJOR 19
+!define VERSION_MINOR 00
+!define VERSION_POSTFIX_FULL ""
 !ifdef WIN64
 !ifdef IA64
 !define VERSION_SYS_POSTFIX_FULL " for Windows IA-64"
