diff --git a/DOC/7zip.wxs b/DOC/7zip.wxs
index 33b677e..338d582 100644
--- a/DOC/7zip.wxs
+++ b/DOC/7zip.wxs
@@ -1,7 +1,7 @@ Hunk #1, a/DOC/7zip.wxs
 <?xml version="1.0"?>
 
-<?define VerMajor = "17" ?>
-<?define VerMinor = "01" ?>
+<?define VerMajor = "19" ?>
+<?define VerMinor = "00" ?>
 <?define VerBuild = "00" ?>
 <?define MmVer = "$(var.VerMajor).$(var.VerMinor)" ?>
 <?define MmHex = "$(var.VerMajor)$(var.VerMinor)" ?>
