diff --git a/DOC/7zip.wxs b/DOC/7zip.wxs
index 33b677e..338d582 100644
--- a/DOC/7zip.wxs
+++ b/DOC/7zip.wxs
@@ -294,6 +294,7 @@ Hunk #3, a/DOC/7zip.wxs
               <File Id="ja.txt" Name="ja.txt" />
               <File Id="ka.txt" Name="ka.txt" />
               <File Id="kaa.txt" Name="kaa.txt" />
+              <File Id="kab.txt" Name="kab.txt" />
               <File Id="kk.txt" Name="kk.txt" />
               <File Id="ko.txt" Name="ko.txt" />
               <File Id="ku.txt" Name="ku.txt" />
