diff --git a/C/LzmaEnc.c b/C/LzmaEnc.c
index 9b7e691..46a0db0 100644
--- a/C/LzmaEnc.c
+++ b/C/LzmaEnc.c
@@ -462,9 +462,10 @@ Hunk #13, a/C/LzmaEnc.c SRes LzmaEnc_SetProps(CLzmaEncHandle pp, const CLzmaEncProps *props2)
   p->lp = props.lp;
   p->pb = props.pb;
   p->fastMode = (props.algo == 0);
+  // p->_maxMode = True;
   p->matchFinderBase.btMode = (Byte)(props.btMode ? 1 : 0);
   {
-    UInt32 numHashBytes = 4;
+    unsigned numHashBytes = 4;
     if (props.btMode)
     {
       if (props.numHashBytes < 2)
