diff --git a/C/MtCoder.c b/C/MtCoder.c
index 32bd0fa..9535985 100644
--- a/C/MtCoder.c
+++ b/C/MtCoder.c
@@ -298,7 +298,7 @@ Hunk #3, a/C/MtCoder.c static SRes ThreadFunc2(CMtCoderThread *t)
       res = mtc->mtCallback->Code(mtc->mtCallbackObject, t->index, bufIndex,
           mtc->inStream ? t->inBuf : inData, size, finished);
       
-      MtProgress_Reinit(&mtc->mtProgress, t->index);
+      // MtProgress_Reinit(&mtc->mtProgress, t->index);
 
       if (res != SZ_OK)
         MtProgress_SetError(&mtc->mtProgress, res);
