diff --git a/C/MtCoder.c b/C/MtCoder.c
index 32bd0fa..9535985 100644
--- a/C/MtCoder.c
+++ b/C/MtCoder.c
@@ -604,7 +604,7 @@ Hunk #5, a/C/MtCoder.c SRes MtCoder_Code(CMtCoder *p)
       {
         const CMtCoderBlock *block = &p->blocks[bi];
         unsigned bufIndex = block->bufIndex;
-        Bool finished = block->finished;
+        BoolInt finished = block->finished;
         if (res == SZ_OK && block->res != SZ_OK)
           res = block->res;
 
