diff --git a/C/MtCoder.h b/C/MtCoder.h
index 624aff2..5a5f4d1 100644
--- a/C/MtCoder.h
+++ b/C/MtCoder.h
@@ -24,33 +24,20 @@ Hunk #2, a/C/MtCoder.h EXTERN_C_BEGIN
 #endif
 
 
-typedef struct
-{
-  UInt64 inSize;
-  UInt64 outSize;
-} CMtProgressSizes;
-
-
-typedef struct
-{
-  ICompressProgress *progress;
-  SRes res;
-  UInt64 totalInSize;
-  UInt64 totalOutSize;
-  CCriticalSection cs;
-  CMtProgressSizes sizes[MTCODER__THREADS_MAX];
-} CMtProgress;
+#ifndef _7ZIP_ST
 
 
 typedef struct
 {
   ICompressProgress vt;
   CMtProgress *mtProgress;
-  unsigned index;
+  UInt64 inSize;
+  UInt64 outSize;
 } CMtProgressThunk;
 
 void MtProgressThunk_CreateVTable(CMtProgressThunk *p);
     
+#define MtProgressThunk_Init(p) { (p)->inSize = 0; (p)->outSize = 0; }
 
 
 struct _CMtCoder;
