diff --git a/C/Ppmd7.h b/C/Ppmd7.h
index 45002d9..610539a 100644
--- a/C/Ppmd7.h
+++ b/C/Ppmd7.h
@@ -60,7 +60,7 @@ Hunk #2, a/C/Ppmd7.h typedef struct
 } CPpmd7;
 
 void Ppmd7_Construct(CPpmd7 *p);
-Bool Ppmd7_Alloc(CPpmd7 *p, UInt32 size, ISzAllocPtr alloc);
+BoolInt Ppmd7_Alloc(CPpmd7 *p, UInt32 size, ISzAllocPtr alloc);
 void Ppmd7_Free(CPpmd7 *p, ISzAllocPtr alloc);
 void Ppmd7_Init(CPpmd7 *p, unsigned maxOrder);
 #define Ppmd7_WasAllocated(p) ((p)->Base != NULL)
