diff --git a/C/Ppmd7.h b/C/Ppmd7.h
index 45002d9..610539a 100644
--- a/C/Ppmd7.h
+++ b/C/Ppmd7.h
@@ -115,7 +115,7 @@ Hunk #3, a/C/Ppmd7.h typedef struct
 } CPpmd7z_RangeDec;
 
 void Ppmd7z_RangeDec_CreateVTable(CPpmd7z_RangeDec *p);
-Bool Ppmd7z_RangeDec_Init(CPpmd7z_RangeDec *p);
+BoolInt Ppmd7z_RangeDec_Init(CPpmd7z_RangeDec *p);
 #define Ppmd7z_RangeDec_IsFinishedOK(p) ((p)->Code == 0)
 
 int Ppmd7_DecodeSymbol(CPpmd7 *p, const IPpmd7_RangeDec *rc);
