diff --git a/C/Ppmd8.h b/C/Ppmd8.h
index 3b15920..51c497d 100644
--- a/C/Ppmd8.h
+++ b/C/Ppmd8.h
@@ -86,7 +86,7 @@ Hunk #2, a/C/Ppmd8.h typedef struct
 } CPpmd8;
 
 void Ppmd8_Construct(CPpmd8 *p);
-Bool Ppmd8_Alloc(CPpmd8 *p, UInt32 size, ISzAllocPtr alloc);
+BoolInt Ppmd8_Alloc(CPpmd8 *p, UInt32 size, ISzAllocPtr alloc);
 void Ppmd8_Free(CPpmd8 *p, ISzAllocPtr alloc);
 void Ppmd8_Init(CPpmd8 *p, unsigned maxOrder, unsigned restoreMethod);
 #define Ppmd8_WasAllocated(p) ((p)->Base != NULL)
