diff --git a/C/Util/7z/7zMain.c b/C/Util/7z/7zMain.c
index cb60a9b..6ccc830 100644
--- a/C/Util/7z/7zMain.c
+++ b/C/Util/7z/7zMain.c
@@ -202,7 +202,7 @@ Hunk #3, a/C/Util/7z/7zMain.c static WRes MyCreateDir(const UInt16 *name)
 {
   #ifdef USE_WINDOWS_FILE
   
-  return CreateDirectoryW(name, NULL) ? 0 : GetLastError();
+  return CreateDirectoryW((LPCWSTR)name, NULL) ? 0 : GetLastError();
   
   #else
 
