diff --git a/C/Util/7z/7zMain.c b/C/Util/7z/7zMain.c
index cb60a9b..6ccc830 100644
--- a/C/Util/7z/7zMain.c
+++ b/C/Util/7z/7zMain.c
@@ -622,7 +622,7 @@ Hunk #8, a/C/Util/7z/7zMain.c int MY_CDECL main(int numargs, char *args[])
                We remove posix bits, if we detect posix mode field */
             if ((attrib & 0xF0000000) != 0)
               attrib &= 0x7FFF;
-            SetFileAttributesW(destPath, attrib);
+            SetFileAttributesW((LPCWSTR)destPath, attrib);
           }
           #endif
         }
