diff --git a/C/Util/7zipInstall/7zipInstall.c b/C/Util/7zipInstall/7zipInstall.c
index db09a32..2022e2f 100644
--- a/C/Util/7zipInstall/7zipInstall.c
+++ b/C/Util/7zipInstall/7zipInstall.c
@@ -135,11 +135,11 @@ Hunk #4, a/C/Util/7zipInstall/7zipInstall.c static DWORD GetFileVersion(LPCWSTR s)
   if (!vi)
     return 0;
   
-  if (GetFileVersionInfoW(s, 0, size, vi))
+  if (my_GetFileVersionInfoW(s, 0, size, vi))
   {
     VS_FIXEDFILEINFO *fi = NULL;
     UINT fiLen = 0;
-    if (VerQueryValueW(vi, L"\\", (LPVOID *)&fi, &fiLen))
+    if (my_VerQueryValueW(vi, L"\\", (LPVOID *)&fi, &fiLen))
       version = fi->dwFileVersionMS;
   }
   
