diff --git a/C/Util/7zipUninstall/7zipUninstall.c b/C/Util/7zipUninstall/7zipUninstall.c
index 0bf27cc..1676117 100644
--- a/C/Util/7zipUninstall/7zipUninstall.c
+++ b/C/Util/7zipUninstall/7zipUninstall.c
@@ -300,7 +300,7 @@ Hunk #8, a/C/Util/7zipUninstall/7zipUninstall.c static void SetShellProgramsGroup(HWND hwndOwner)
     {
       const size_t baseLen = wcslen(link);
       unsigned k;
-      Bool needDelete = False;
+      BoolInt needDelete = False;
 
       for (k = 0; k < 2; k++)
       {
