diff --git a/C/Util/7zipUninstall/7zipUninstall.c b/C/Util/7zipUninstall/7zipUninstall.c
index 0bf27cc..1676117 100644
--- a/C/Util/7zipUninstall/7zipUninstall.c
+++ b/C/Util/7zipUninstall/7zipUninstall.c
@@ -347,7 +347,7 @@ Hunk #9, a/C/Util/7zipUninstall/7zipUninstall.c static const WCHAR * const k_AppPaths_7zFm = L"Software\\Microsoft\\Windows\\Cur
 static const WCHAR * const k_Uninstall_7zip = k_REG_Uninstall L"7-Zip";
 
 
-static Bool AreEqual_Path_PrefixName(const wchar_t *s, const wchar_t *prefix, const wchar_t *name)
+static BoolInt AreEqual_Path_PrefixName(const wchar_t *s, const wchar_t *prefix, const wchar_t *name)
 {
   if (!IsString1PrefixedByString2_NoCase(s, prefix))
     return False;
