diff --git a/C/Util/7zipUninstall/7zipUninstall.c b/C/Util/7zipUninstall/7zipUninstall.c
index 0bf27cc..1676117 100644
--- a/C/Util/7zipUninstall/7zipUninstall.c
+++ b/C/Util/7zipUninstall/7zipUninstall.c
@@ -491,7 +491,7 @@ Hunk #12, a/C/Util/7zipUninstall/7zipUninstall.c static BOOL RemoveFileAfterReboot()
 
 #define IS_LIMIT_CHAR(c) (c == 0 || c == ' ')
 
-static Bool IsThereSpace(const wchar_t *s)
+static BoolInt IsThereSpace(const wchar_t *s)
 {
   for (;;)
   {
