diff --git a/C/Util/7zipUninstall/7zipUninstall.c b/C/Util/7zipUninstall/7zipUninstall.c
index 0bf27cc..1676117 100644
--- a/C/Util/7zipUninstall/7zipUninstall.c
+++ b/C/Util/7zipUninstall/7zipUninstall.c
@@ -805,7 +805,7 @@ Hunk #18, a/C/Util/7zipUninstall/7zipUninstall.c int APIENTRY WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance,
     lpCmdLine, int nCmdShow)
 {
   const wchar_t *cmdParams;
-  Bool useTemp = True;
+  BoolInt useTemp = True;
 
   UNUSED_VAR(hPrevInstance)
   UNUSED_VAR(lpCmdLine)
