diff --git a/C/Xz.h b/C/Xz.h
index 285fb55..544ee18 100644
--- a/C/Xz.h
+++ b/C/Xz.h
@@ -139,6 +139,9 @@ Hunk #3, a/C/Xz.h SRes Xzs_ReadBackward(CXzs *p, ILookInStream *inStream, Int64 *startOffset, ICom
 UInt64 Xzs_GetNumBlocks(const CXzs *p);
 UInt64 Xzs_GetUnpackSize(const CXzs *p);
 
+
+// ECoderStatus values are identical to ELzmaStatus values of LZMA2 decoder
+
 typedef enum
 {
   CODER_STATUS_NOT_SPECIFIED,               /* use main error code instead */
