diff --git a/C/XzDec.c b/C/XzDec.c
index 67216a5..395e83f 100644
--- a/C/XzDec.c
+++ b/C/XzDec.c
@@ -522,6 +522,33 @@ Hunk #14, a/C/XzDec.c static Bool Xz_CheckFooter(CXzStreamFlags flags, UInt64 indexSize, const Byte *b
   if (s == 0) return SZ_ERROR_ARCHIVE; pos += s; }
 
 
+static BoolInt XzBlock_AreSupportedFilters(const CXzBlock *p)
+{
+  unsigned numFilters = XzBlock_GetNumFilters(p) - 1;
+  unsigned i;
+  {
+    const CXzFilter *f = &p->filters[numFilters];
+    if (f->id != XZ_ID_LZMA2 || f->propsSize != 1 || f->props[0] > 40)
+      return False;
+  }
+
+  for (i = 0; i < numFilters; i++)
+  {
+    const CXzFilter *f = &p->filters[i];
+    if (f->id == XZ_ID_Delta)
+    {
+      if (f->propsSize != 1)
+        return False;
+    }
+    else if (f->id < XZ_ID_Delta
+        || f->id > XZ_ID_SPARC
+        || (f->propsSize != 0 && f->propsSize != 4))
+      return False;
+  }
+  return True;
+}
+
+
 SRes XzBlock_Parse(CXzBlock *p, const Byte *header)
 {
   unsigned pos;
