diff --git a/C/Bcj2.c b/C/Bcj2.c
index 0efff4e..9a0046a 100644
--- a/C/Bcj2.c
+++ b/C/Bcj2.c
@@ -232,10 +232,10 @@ Hunk #2, a/C/Bcj2.c SRes Bcj2Dec_Decode(CBcj2Dec *p)
       
       if (rem < 4)
       {
-        SizeT i;
-        SetUi32(p->temp, val);
-        for (i = 0; i < rem; i++)
-          dest[i] = p->temp[i];
+        p->temp[0] = (Byte)val; if (rem > 0) dest[0] = (Byte)val; val >>= 8;
+        p->temp[1] = (Byte)val; if (rem > 1) dest[1] = (Byte)val; val >>= 8;
+        p->temp[2] = (Byte)val; if (rem > 2) dest[2] = (Byte)val; val >>= 8;
+        p->temp[3] = (Byte)val;
         p->dest = dest + rem;
         p->state = BCJ2_DEC_STATE_ORIG_0 + (unsigned)rem;
         break;
