diff --git a/C/XzDec.c b/C/XzDec.c
index 67216a5..395e83f 100644
--- a/C/XzDec.c
+++ b/C/XzDec.c
@@ -669,12 +669,17 @@ Hunk #19, a/C/XzDec.c SRes XzUnpacker_Code(CXzUnpacker *p, Byte *dest, SizeT *destLen,
       SRes res;
 
       ECoderFinishMode finishMode2 = finishMode;
+      BoolInt srcFinished2 = srcFinished;
+      BoolInt destFinish = False;
 
       if (p->block.packSize != (UInt64)(Int64)-1)
       {
         UInt64 rem = p->block.packSize - p->packSize;
-        if (srcLen2 > rem)
+        if (srcLen2 >= rem)
+        {
+          srcFinished2 = True;
           srcLen2 = (SizeT)rem;
+        }
         if (rem == 0 && p->block.unpackSize == p->unpackSize)
           return SZ_ERROR_DATA;
       }
