diff --git a/C/XzDec.c b/C/XzDec.c
index 67216a5..395e83f 100644
--- a/C/XzDec.c
+++ b/C/XzDec.c
@@ -697,15 +697,23 @@ Hunk #21, a/C/XzDec.c SRes XzUnpacker_Code(CXzUnpacker *p, Byte *dest, SizeT *destLen,
       }
       */
       
-      res = MixCoder_Code(&p->decoder, dest, &destLen2, src, &srcLen2, False, finishMode2, status);
-      XzCheck_Update(&p->check, dest, destLen2);
+      {
+        res = MixCoder_Code(&p->decoder,
+            (p->outBuf ? NULL : dest), &destLen2, destFinish,
+            src, &srcLen2, srcFinished2,
+            finishMode2);
+        
+        *status = p->decoder.status;
+        XzCheck_Update(&p->check, (p->outBuf ? p->outBuf + p->outDataWritten : dest), destLen2);
+        if (!p->outBuf)
+          dest += destLen2;
+        p->outDataWritten += destLen2;
+      }
       
       (*srcLen) += srcLen2;
       src += srcLen2;
       p->packSize += srcLen2;
-      
       (*destLen) += destLen2;
-      dest += destLen2;
       p->unpackSize += destLen2;
 
       RINOK(res);
