diff --git a/C/XzDec.c b/C/XzDec.c
index 67216a5..395e83f 100644
--- a/C/XzDec.c
+++ b/C/XzDec.c
@@ -756,6 +756,8 @@ Hunk #23, a/C/XzDec.c SRes XzUnpacker_Code(CXzUnpacker *p, Byte *dest, SizeT *destLen,
         {
           if (p->pos < XZ_SIG_SIZE && *src != XZ_SIG[p->pos])
             return SZ_ERROR_NO_ARCHIVE;
+          if (p->decodeToStreamSignature)
+            return SZ_OK;
           p->buf[p->pos++] = *src++;
           (*srcLen)++;
         }
