diff --git a/C/XzDec.c b/C/XzDec.c
index 67216a5..395e83f 100644
--- a/C/XzDec.c
+++ b/C/XzDec.c
@@ -808,20 +808,32 @@ Hunk #24, a/C/XzDec.c SRes XzUnpacker_Code(CXzUnpacker *p, Byte *dest, SizeT *destLen,
         else
         {
           RINOK(XzBlock_Parse(&p->block, p->buf));
+          if (!XzBlock_AreSupportedFilters(&p->block))
+            return SZ_ERROR_UNSUPPORTED;
           p->numTotalBlocks++;
           p->state = XZ_STATE_BLOCK;
           p->packSize = 0;
           p->unpackSize = 0;
           XzCheck_Init(&p->check, XzFlags_GetCheckType(p->streamFlags));
-          RINOK(XzDec_Init(&p->decoder, &p->block));
+          if (p->parseMode)
+          {
+            p->headerParsedOk = True;
+            return SZ_OK;
+          }
+          RINOK(XzDecMix_Init(&p->decoder, &p->block, p->outBuf, p->outBufSize));
         }
         break;
       }
 
       case XZ_STATE_BLOCK_FOOTER:
       {
-        if (((p->packSize + p->alignPos) & 3) != 0)
+        if ((((unsigned)p->packSize + p->alignPos) & 3) != 0)
         {
+          if (srcRem == 0)
+          {
+            *status = CODER_STATUS_NEEDS_MORE_INPUT;
+            return SZ_OK;
+          }
           (*srcLen)++;
           p->alignPos++;
           if (*src++ != 0)
