diff --git a/C/XzDec.c b/C/XzDec.c
index 67216a5..395e83f 100644
--- a/C/XzDec.c
+++ b/C/XzDec.c
@@ -833,6 +833,11 @@ Hunk #25, a/C/XzDec.c SRes XzUnpacker_Code(CXzUnpacker *p, Byte *dest, SizeT *destLen,
           UInt32 cur = checkSize - p->pos;
           if (cur != 0)
           {
+            if (srcRem == 0)
+            {
+              *status = CODER_STATUS_NEEDS_MORE_INPUT;
+              return SZ_OK;
+            }
             if (cur > srcRem)
               cur = (UInt32)srcRem;
             memcpy(p->buf + p->pos, src, cur);
