diff --git a/C/XzEnc.c b/C/XzEnc.c
index 2b95a0e..d0a8b44 100644
--- a/C/XzEnc.c
+++ b/C/XzEnc.c
@@ -362,13 +362,16 @@ Hunk #4, a/C/XzEnc.c static SRes SeqInFilter_Read(const ISeqInStream *pp, void *data, size_t *size)
     }
     {
       SizeT srcLen = p->endPos - p->curPos;
-      int wasFinished;
+      ECoderStatus status;
       SRes res;
       *size = sizeOriginal;
-      res = p->StateCoder.Code(p->StateCoder.p, data, size, p->buf + p->curPos, &srcLen,
-        p->srcWasFinished, CODER_FINISH_ANY, &wasFinished);
+      res = p->StateCoder.Code2(p->StateCoder.p,
+          (Byte *)data, size,
+          p->buf + p->curPos, &srcLen,
+          p->srcWasFinished, CODER_FINISH_ANY,
+          &status);
       p->curPos += srcLen;
-      if (*size != 0 || srcLen == 0 || res != 0)
+      if (*size != 0 || srcLen == 0 || res != SZ_OK)
         return res;
     }
   }
