diff --git a/C/XzEnc.c b/C/XzEnc.c
index 2b95a0e..d0a8b44 100644
--- a/C/XzEnc.c
+++ b/C/XzEnc.c
@@ -1063,7 +1063,7 @@ Hunk #9, a/C/XzEnc.c static SRes XzEnc_MtCallback_Code(void *pp, unsigned coderIndex, unsigned outBuf
   
   MtProgressThunk_CreateVTable(&progressThunk);
   progressThunk.mtProgress = &me->mtCoder.mtProgress;
-  progressThunk.index = coderIndex;
+  MtProgressThunk_Init(&progressThunk);
 
   {
     CXzEncBlockInfo blockSizes;
