diff --git a/C/XzEnc.c b/C/XzEnc.c
index 2b95a0e..d0a8b44 100644
--- a/C/XzEnc.c
+++ b/C/XzEnc.c
@@ -1245,6 +1245,8 @@ Hunk #11, a/C/XzEnc.c SRes XzEnc_Encode(CXzEncHandle pp, ISeqOutStream *outStream, ISeqInStream *inStr
           && props->reduceSize >= progress2.inOffset)
         rem = props->reduceSize - progress2.inOffset;
       */
+
+      blockSizes.headerSize = 0; // for GCC
       
       RINOK(Xz_CompressBlock(
           &p->lzmaf_Items[0],
