diff --git a/C/XzIn.c b/C/XzIn.c
index 357549e..ff48e2d 100644
--- a/C/XzIn.c
+++ b/C/XzIn.c
@@ -103,7 +103,7 @@ Hunk #3, a/C/XzIn.c static SRes Xz_ReadIndex2(CXzStream *p, const Byte *buf, size_t size, ISzAllocPt
   {
     size_t i;
     p->numBlocks = numBlocks;
-    p->blocks = ISzAlloc_Alloc(alloc, sizeof(CXzBlockSizes) * numBlocks);
+    p->blocks = (CXzBlockSizes *)ISzAlloc_Alloc(alloc, sizeof(CXzBlockSizes) * numBlocks);
     if (!p->blocks)
       return SZ_ERROR_MEM;
     for (i = 0; i < numBlocks; i++)
