diff --git a/CPP/7zip/Aes.mak b/CPP/7zip/Aes.mak
index fbd70aa..20f1a72 100644
--- a/CPP/7zip/Aes.mak
+++ b/CPP/7zip/Aes.mak
@@ -1,7 +1,7 @@ Hunk #1, a/CPP/7zip/Aes.mak
 C_OBJS = $(C_OBJS) \
   $O\Aes.obj
 
-!IF "$(CPU)" != "IA64" && "$(CPU)" != "MIPS" && "$(CPU)" != "ARM" && "$(CPU)" != "ARM64"
+!IF "$(PLATFORM)" != "ia64" && "$(PLATFORM)" != "mips" && "$(PLATFORM)" != "arm" && "$(PLATFORM)" != "arm64"
 ASM_OBJS = $(ASM_OBJS) \
   $O\AesOpt.obj
 !ENDIF
