diff --git a/C/Bcj2Enc.c b/C/Bcj2Enc.c
index 98b775b..bfbeb8e 100644
--- a/C/Bcj2Enc.c
+++ b/C/Bcj2Enc.c
@@ -53,7 +53,7 @@ Hunk #3, a/C/Bcj2Enc.c void Bcj2Enc_Init(CBcj2Enc *p)
     p->probs[i] = kBitModelTotal >> 1;
 }
 
-static Bool MY_FAST_CALL RangeEnc_ShiftLow(CBcj2Enc *p)
+static BoolInt MY_FAST_CALL RangeEnc_ShiftLow(CBcj2Enc *p)
 {
   if ((UInt32)p->low < (UInt32)0xFF000000 || (UInt32)(p->low >> 32) != 0)
   {
