diff --git a/C/Bcj2Enc.c b/C/Bcj2Enc.c
index 98b775b..bfbeb8e 100644
--- a/C/Bcj2Enc.c
+++ b/C/Bcj2Enc.c
@@ -166,7 +166,7 @@ Hunk #4, a/C/Bcj2Enc.c static void Bcj2Enc_Encode_2(CBcj2Enc *p)
  
         {
           Byte context = (Byte)(num == 0 ? p->prevByte : src[-1]);
-          Bool needConvert;
+          BoolInt needConvert;
 
           p->bufs[BCJ2_STREAM_MAIN] = dest + 1;
           p->ip += (UInt32)num + 1;
