diff --git a/CPP/7zip/Archive/7z/7zHandlerOut.cpp b/CPP/7zip/Archive/7z/7zHandlerOut.cpp
index c4fabed..b6be06a 100644
--- a/CPP/7zip/Archive/7z/7zHandlerOut.cpp
+++ b/CPP/7zip/Archive/7z/7zHandlerOut.cpp
@@ -263,6 +263,9 @@ Hunk #3, a/CPP/7zip/Archive/7z/7zHandlerOut.cpp STDMETHODIMP CHandler::UpdateItems(ISequentialOutStream *outStream, UInt32 numIt
     db = &_db;
   #endif
 
+  if (db && !db->CanUpdate())
+    return E_NOTIMPL;
+
   /*
   CMyComPtr<IArchiveGetRawProps> getRawProps;
   updateCallback->QueryInterface(IID_IArchiveGetRawProps, (void **)&getRawProps);
