diff --git a/CPP/7zip/Archive/7z/7zIn.h b/CPP/7zip/Archive/7z/7zIn.h
index 2d5fbb7..6a61d31 100644
--- a/CPP/7zip/Archive/7z/7zIn.h
+++ b/CPP/7zip/Archive/7z/7zIn.h
@@ -257,6 +257,16 @@ Hunk #1, a/CPP/7zip/Archive/7z/7zIn.h struct CDbEx: public CDatabase
     PhySize = 0;
   }
 
+  bool CanUpdate() const
+  {
+    if (ThereIsHeaderError
+        || UnexpectedEnd
+        || StartHeaderWasRecovered
+        || UnsupportedFeatureError)
+      return false;
+    return true;
+  }
+
   void FillLinks();
   
   UInt64 GetFolderStreamPos(CNum folderIndex, unsigned indexInFolder) const
