diff --git a/C/BwtSort.c b/C/BwtSort.c
index aa31920..cc2f4b2 100644
--- a/C/BwtSort.c
+++ b/C/BwtSort.c
@@ -448,7 +448,7 @@ Hunk #2, a/C/BwtSort.c UInt32 BlockSort(UInt32 *Indices, const Byte *data, UInt32 blockSize)
 
       groupSize = ((Indices[i] & ~0xC0000000) >> kNumBitsMax);
       {
-      Bool finishedGroup = ((Indices[i] & 0x80000000) == 0);
+      BoolInt finishedGroup = ((Indices[i] & 0x80000000) == 0);
       if ((Indices[i] & 0x40000000) != 0)
       {
         groupSize += ((Indices[(size_t)i + 1] >> kNumBitsMax) << kNumExtra0Bits);
