diff --git a/CPP/7zip/Archive/ArHandler.cpp b/CPP/7zip/Archive/ArHandler.cpp
index 816a84e..09a6220 100644
--- a/CPP/7zip/Archive/ArHandler.cpp
+++ b/CPP/7zip/Archive/ArHandler.cpp
@@ -170,8 +170,8 @@ Hunk #1, a/CPP/7zip/Archive/ArHandler.cpp static bool OctalToNumber32(const char *s, unsigned size, UInt32 &res)
   res = 0;
   char sz[32];
   size = RemoveTailSpaces(sz, s, size);
-  if (size == 0)
-    return true; // some items doesn't contaion any numbers
+  if (size == 0 || strcmp(sz, "-1") == 0)
+    return true; // some items don't contain any numbers
   const char *end;
   UInt64 res64 = ConvertOctStringToUInt64(sz, &end);
   if ((unsigned)(end - sz) != size)
