diff --git a/CPP/7zip/Archive/ArHandler.cpp b/CPP/7zip/Archive/ArHandler.cpp
index 816a84e..09a6220 100644
--- a/CPP/7zip/Archive/ArHandler.cpp
+++ b/CPP/7zip/Archive/ArHandler.cpp
@@ -185,8 +185,8 @@ Hunk #2, a/CPP/7zip/Archive/ArHandler.cpp static bool DecimalToNumber(const char *s, unsigned size, UInt64 &res)
   res = 0;
   char sz[32];
   size = RemoveTailSpaces(sz, s, size);
-  if (size == 0)
-    return true; // some items doesn't contaion any numbers
+  if (size == 0 || strcmp(sz, "-1") == 0)
+    return true; // some items don't contain any numbers
   const char *end;
   res = ConvertStringToUInt64(sz, &end);
   return ((unsigned)(end - sz) == size);
