diff --git a/CPP/7zip/Archive/Bz2Handler.cpp b/CPP/7zip/Archive/Bz2Handler.cpp
index 9842843..b0c2f75 100644
--- a/CPP/7zip/Archive/Bz2Handler.cpp
+++ b/CPP/7zip/Archive/Bz2Handler.cpp
@@ -366,7 +366,13 @@ Hunk #1, a/CPP/7zip/Archive/Bz2Handler.cpp STDMETHODIMP CHandler::UpdateItems(ISequentialOutStream *outStream, UInt32 numIt
         return E_INVALIDARG;
       size = prop.uhVal.QuadPart;
     }
-    return UpdateArchive(size, outStream, _props, updateCallback);
+
+    CMethodProps props2 = _props;
+    #ifndef _7ZIP_ST
+    props2.AddProp_NumThreads(_props._numThreads);
+    #endif
+
+    return UpdateArchive(size, outStream, props2, updateCallback);
   }
 
   if (indexInArchive != 0)
