diff --git a/C/CpuArch.c b/C/CpuArch.c
index 554ffa4..02e482e 100644
--- a/C/CpuArch.c
+++ b/C/CpuArch.c
@@ -115,7 +115,7 @@ Hunk #2, a/C/CpuArch.c void MyCPUID(UInt32 function, UInt32 *a, UInt32 *b, UInt32 *c, UInt32 *d)
   #endif
 }
 
-Bool x86cpuid_CheckAndRead(Cx86cpuid *p)
+BoolInt x86cpuid_CheckAndRead(Cx86cpuid *p)
 {
   CHECK_CPUID_IS_SUPPORTED
   MyCPUID(0, &p->maxFunc, &p->vendor[0], &p->vendor[2], &p->vendor[1]);
