diff --git a/CPP/7zip/Archive/ElfHandler.cpp b/CPP/7zip/Archive/ElfHandler.cpp
index 6267477..c5ce279 100644
--- a/CPP/7zip/Archive/ElfHandler.cpp
+++ b/CPP/7zip/Archive/ElfHandler.cpp
@@ -567,6 +567,7 @@ Hunk #1, a/CPP/7zip/Archive/ElfHandler.cpp static const char * const g_Machines[] =
 
 static const CUInt32PCharPair g_MachinePairs[] =
 {
+  { 243, "RISC-V" },
   { 47787, "Xilinx MicroBlaze" }
   // { 0x9026, "Alpha" }
 };
