diff --git a/CPP/7zip/Archive/ExtHandler.cpp b/CPP/7zip/Archive/ExtHandler.cpp
index d8417fe..db65fbe 100644
--- a/CPP/7zip/Archive/ExtHandler.cpp
+++ b/CPP/7zip/Archive/ExtHandler.cpp
@@ -1596,6 +1596,17 @@ Hunk #1, a/CPP/7zip/Archive/ExtHandler.cpp STDMETHODIMP CHandler::Close()
 }
 
 
+static void ChangeSeparatorsInName(char *s, unsigned num)
+{
+  for (unsigned i = 0; i < num; i++)
+  {
+    char c = s[i];
+    if (c == CHAR_PATH_SEPARATOR || c == '/')
+      s[i] = '_';
+  }
+}
+
+
 void CHandler::GetPath(unsigned index, AString &s) const
 {
   s.Empty();
