diff --git a/CPP/7zip/Archive/FatHandler.cpp b/CPP/7zip/Archive/FatHandler.cpp
index 9c37c06..bf6053e 100644
--- a/CPP/7zip/Archive/FatHandler.cpp
+++ b/CPP/7zip/Archive/FatHandler.cpp
@@ -732,7 +732,13 @@ Hunk #2, a/CPP/7zip/Archive/FatHandler.cpp HRESULT CDatabase::Open()
   RINOK(OpenProgressFat());
 
   if ((Fat[0] & 0xFF) != Header.MediaType)
-     return S_FALSE;
+  {
+    // that case can mean error in FAT,
+    // but xdf file: (MediaType == 0xF0 && Fat[0] == 0xFF9)
+    // 19.00: so we use non-strict check
+    if ((Fat[0] & 0xFF) < 0xF0)
+      return S_FALSE;
+  }
 
   RINOK(ReadDir(-1, Header.RootCluster, 0));
 
