diff --git a/CPP/7zip/Archive/HfsHandler.cpp b/CPP/7zip/Archive/HfsHandler.cpp
index d02f072..ca1d7bd 100644
--- a/CPP/7zip/Archive/HfsHandler.cpp
+++ b/CPP/7zip/Archive/HfsHandler.cpp
@@ -317,11 +317,15 @@ Hunk #1, a/CPP/7zip/Archive/HfsHandler.cpp class CDatabase
   // bool CaseSensetive;
   UString ResFileName;
 
+  UInt64 SpecOffset;
   UInt64 PhySize;
+  UInt64 PhySize2;
 
   void Clear()
   {
+    SpecOffset = 0;
     PhySize = 0;
+    PhySize2 = 0;
     HeadersError = false;
     ThereAreAltStreams = false;
     // CaseSensetive = false;
