diff --git a/CPP/7zip/Archive/HfsHandler.cpp b/CPP/7zip/Archive/HfsHandler.cpp
index d02f072..ca1d7bd 100644
--- a/CPP/7zip/Archive/HfsHandler.cpp
+++ b/CPP/7zip/Archive/HfsHandler.cpp
@@ -1330,7 +1330,14 @@ Hunk #6, a/CPP/7zip/Archive/HfsHandler.cpp STDMETHODIMP CHandler::GetArchiveProperty(PROPID propID, PROPVARIANT *value)
   {
     case kpidExtension: prop = Header.IsHfsX() ? "hfsx" : "hfs"; break;
     case kpidMethod: prop = Header.IsHfsX() ? "HFSX" : "HFS+"; break;
-    case kpidPhySize: prop = PhySize; break;
+    case kpidPhySize:
+    {
+      UInt64 v = SpecOffset + PhySize;
+      if (v < PhySize2)
+        v = PhySize2;
+      prop = v;
+      break;
+    }
     case kpidClusterSize: prop = (UInt32)1 << Header.BlockSizeLog; break;
     case kpidFreeSpace: prop = (UInt64)Header.GetFreeSize(); break;
     case kpidMTime: HfsTimeToProp(Header.MTime, prop); break;
