diff --git a/CPP/7zip/Archive/HfsHandler.cpp b/CPP/7zip/Archive/HfsHandler.cpp
index d02f072..ca1d7bd 100644
--- a/CPP/7zip/Archive/HfsHandler.cpp
+++ b/CPP/7zip/Archive/HfsHandler.cpp
@@ -1865,6 +1865,7 @@ Hunk #8, a/CPP/7zip/Archive/HfsHandler.cpp STDMETHODIMP CHandler::GetStream(UInt32 index, ISequentialInStream **stream)
 }
 
 static const Byte k_Signature[] = {
+    2, 'B', 'D',
     4, 'H', '+', 0, 4,
     4, 'H', 'X', 0, 5 };
 
