diff --git a/C/CpuArch.c b/C/CpuArch.c
index 554ffa4..02e482e 100644
--- a/C/CpuArch.c
+++ b/C/CpuArch.c
@@ -197,4 +197,22 @@ Hunk #6, a/C/CpuArch.c Bool CPU_Is_Aes_Supported()
   return (p.c >> 25) & 1;
 }
 
+BoolInt CPU_IsSupported_PageGB()
+{
+  Cx86cpuid cpuid;
+  if (!x86cpuid_CheckAndRead(&cpuid))
+    return False;
+  {
+    UInt32 d[4] = { 0 };
+    MyCPUID(0x80000000, &d[0], &d[1], &d[2], &d[3]);
+    if (d[0] < 0x80000001)
+      return False;
+  }
+  {
+    UInt32 d[4] = { 0 };
+    MyCPUID(0x80000001, &d[0], &d[1], &d[2], &d[3]);
+    return (d[3] >> 26) & 1;
+  }
+}
+
 #endif
