diff --git a/CPP/7zip/Archive/NtfsHandler.cpp b/CPP/7zip/Archive/NtfsHandler.cpp
index 0eebf00..2137203 100644
--- a/CPP/7zip/Archive/NtfsHandler.cpp
+++ b/CPP/7zip/Archive/NtfsHandler.cpp
@@ -1575,7 +1575,7 @@ Hunk #5, a/CPP/7zip/Archive/NtfsHandler.cpp void CDatabase::GetItemPath(unsigned index, NCOM::CPropVariant &path) const
         if (len != 0)
         {
           size -= len;
-          MyStringCopy(s + size, name.GetRawPtr());
+          CopyName(s + size, name.GetRawPtr());
         }
         s[size + len] = WCHAR_PATH_SEPARATOR;
         continue;
