diff --git a/CPP/7zip/Archive/PeHandler.cpp b/CPP/7zip/Archive/PeHandler.cpp
index 30a6084..675293b 100644
--- a/CPP/7zip/Archive/PeHandler.cpp
+++ b/CPP/7zip/Archive/PeHandler.cpp
@@ -41,9 +41,8 @@ Hunk #1, a/CPP/7zip/Archive/PeHandler.cpp static const UInt32 k_Signature32 = 0x00004550;
 
 static HRESULT CalcCheckSum(ISequentialInStream *stream, UInt32 size, UInt32 excludePos, UInt32 &res)
 {
-  const UInt32 kBufSizeMax = (UInt32)1 << 16;
-  UInt32 bufSize = MyMin(kBufSizeMax, size);
-  bufSize += (bufSize & 1);
+  const UInt32 kBufSizeMax = (UInt32)1 << 15;
+  UInt32 bufSize = kBufSizeMax;
   CByteBuffer buffer(bufSize);
   Byte *buf = buffer;
   UInt32 sum = 0;
