diff --git a/CPP/7zip/Archive/PeHandler.cpp b/CPP/7zip/Archive/PeHandler.cpp
index 30a6084..675293b 100644
--- a/CPP/7zip/Archive/PeHandler.cpp
+++ b/CPP/7zip/Archive/PeHandler.cpp
@@ -58,9 +58,6 @@ Hunk #2, a/CPP/7zip/Archive/PeHandler.cpp static HRESULT CalcCheckSum(ISequentialInStream *stream, UInt32 size, UInt32 exc
     size_t processed = rem;
     RINOK(ReadStream(stream, buf, &processed));
     
-    if ((processed & 1) != 0)
-      buf[processed] = 0;
-
     for (unsigned j = 0; j < 4; j++)
     {
       UInt32 e = excludePos + j;
