diff --git a/CPP/7zip/Archive/PeHandler.cpp b/CPP/7zip/Archive/PeHandler.cpp
index 30a6084..675293b 100644
--- a/CPP/7zip/Archive/PeHandler.cpp
+++ b/CPP/7zip/Archive/PeHandler.cpp
@@ -720,9 +720,12 @@ Hunk #10, a/CPP/7zip/Archive/PeHandler.cpp class CHandler:
   bool _parseResources;
   bool _checksumError;
 
+  bool IsOpt() const { return _header.OptHeaderSize != 0; }
+
   COptHeader _optHeader;
 
   bool _allowTail;
+  bool _coffMode;
 
   HRESULT LoadDebugSections(IInStream *stream, bool &thereIsSection);
   HRESULT Open2(IInStream *stream, IArchiveOpenCallback *callback);
