diff --git a/CPP/7zip/Archive/PeHandler.cpp b/CPP/7zip/Archive/PeHandler.cpp
index 30a6084..675293b 100644
--- a/CPP/7zip/Archive/PeHandler.cpp
+++ b/CPP/7zip/Archive/PeHandler.cpp
@@ -2333,6 +2333,7 @@ Hunk #23, a/CPP/7zip/Archive/PeHandler.cpp HRESULT CHandler::Open2(IInStream *stream, IArchiveOpenCallback *callback)
     const CSection &sect = _sections[i];
     CMixItem mixItem;
     mixItem.SectionIndex = i;
+    if (IsOpt())
     if (_parseResources && sect.Name == ".rsrc" && _items.IsEmpty())
     {
       const unsigned numMixItems = _mixItems.Size();
