diff --git a/CPP/7zip/Archive/Rar/Rar5Handler.cpp b/CPP/7zip/Archive/Rar/Rar5Handler.cpp
index c422542..320771d 100644
--- a/CPP/7zip/Archive/Rar/Rar5Handler.cpp
+++ b/CPP/7zip/Archive/Rar/Rar5Handler.cpp
@@ -611,6 +611,8 @@ Hunk #5, a/CPP/7zip/Archive/Rar/Rar5Handler.cpp class CInArchive
     UInt64 DataSize;
   };
 
+  CInArchive() {}
+
   HRESULT ReadBlockHeader(CHeader &h);
   bool ReadFileHeader(const CHeader &header, CItem &item);
   void AddToSeekValue(UInt64 addValue)
