diff --git a/C/DllSecur.c b/C/DllSecur.c
index 90352ce..5ea108a 100644
--- a/C/DllSecur.c
+++ b/C/DllSecur.c
@@ -28,10 +28,31 @@ Hunk #2, a/C/DllSecur.c static const char * const g_Dlls =
   "CRYPTBASE\0"
   "OLEACC\0"
   "CLBCATQ\0"
+  "VERSION\0"
   ;
 
 #endif
 
+void My_SetDefaultDllDirectories()
+{
+  #ifndef UNDER_CE
+  
+    OSVERSIONINFO vi;
+    vi.dwOSVersionInfoSize = sizeof(vi);
+    GetVersionEx(&vi);
+    if (!GetVersionEx(&vi) || vi.dwMajorVersion != 6 || vi.dwMinorVersion != 0)
+    {
+      Func_SetDefaultDllDirectories setDllDirs = (Func_SetDefaultDllDirectories)
+          GetProcAddress(GetModuleHandle(TEXT("kernel32.dll")), "SetDefaultDllDirectories");
+      if (setDllDirs)
+        if (setDllDirs(MY_LOAD_LIBRARY_SEARCH_SYSTEM32 | MY_LOAD_LIBRARY_SEARCH_USER_DIRS))
+          return;
+    }
+
+  #endif
+}
+
+
 void LoadSecurityDlls()
 {
   #ifndef UNDER_CE
