diff --git a/CPP/7zip/Archive/Rar/RarHandler.cpp b/CPP/7zip/Archive/Rar/RarHandler.cpp
index c097b15..dd78e31 100644
--- a/CPP/7zip/Archive/Rar/RarHandler.cpp
+++ b/CPP/7zip/Archive/Rar/RarHandler.cpp
@@ -667,7 +667,8 @@ Hunk #5, a/CPP/7zip/Archive/Rar/RarHandler.cpp HRESULT CInArchive::GetNextItem(CItem &item, ICryptoGetTextPassword *getTextPass
         {
           if (processed < offset + 2)
             error = k_ErrorType_Corrupted;
-          ArcInfo.VolNumber = (UInt32)Get16(m_FileHeaderData + offset);
+          else
+            ArcInfo.VolNumber = (UInt32)Get16(m_FileHeaderData + offset);
         }
 
         ArcInfo.EndOfArchive_was_Read = true;
