diff --git a/CPP/7zip/Archive/SwfHandler.cpp b/CPP/7zip/Archive/SwfHandler.cpp
index 16a6e6f..ebb6acf 100644
--- a/CPP/7zip/Archive/SwfHandler.cpp
+++ b/CPP/7zip/Archive/SwfHandler.cpp
@@ -46,7 +46,7 @@ Hunk #1, a/CPP/7zip/Archive/SwfHandler.cpp static const Byte SWF_COMPRESSED_LZMA = 'Z';
 static const Byte SWF_MIN_COMPRESSED_ZLIB_VER = 6;
 static const Byte SWF_MIN_COMPRESSED_LZMA_VER = 13;
 
-static const Byte kVerLim = 20;
+static const Byte kVerLim = 64;
 
 API_FUNC_static_IsArc IsArc_Swf(const Byte *p, size_t size)
 {
