diff --git a/CPP/7zip/Archive/Tar/TarIn.cpp b/CPP/7zip/Archive/Tar/TarIn.cpp
index 760bfa9..3276165 100644
--- a/CPP/7zip/Archive/Tar/TarIn.cpp
+++ b/CPP/7zip/Archive/Tar/TarIn.cpp
@@ -94,7 +94,9 @@ Hunk #1, a/CPP/7zip/Archive/Tar/TarIn.cpp static bool ParseInt64(const char *p, Int64 &val)
 
 static bool ParseInt64_MTime(const char *p, Int64 &val)
 {
-  // rare case tar contains spaces instead of MTime
+  // rare case tar : ZEROs in Docker-Windows TARs
+  // rare case tar : spaces
+  if (GetUi32(p) != 0)
   for (unsigned i = 0; i < 12; i++)
     if (p[i] != ' ')
       return ParseInt64(p, val);
