diff --git a/CPP/7zip/Archive/Tar/TarIn.cpp b/CPP/7zip/Archive/Tar/TarIn.cpp
index 760bfa9..3276165 100644
--- a/CPP/7zip/Archive/Tar/TarIn.cpp
+++ b/CPP/7zip/Archive/Tar/TarIn.cpp
@@ -441,14 +441,14 @@ Hunk #2, a/CPP/7zip/Archive/Tar/TarIn.cpp HRESULT ReadItem(ISequentialInStream *stream, bool &filled, CItemEx &item, EErro
       case 'X':
       {
         // pax Extended Header
-        if (item.Name.IsPrefixedBy("PaxHeader/"))
+        if (item.Name.IsPrefixedBy("PaxHeader/")
+            || item.Name.Find("PaxHeaders.4467/") >= 0)
         {
           RINOK(ReadDataToString(stream, item, pax, error));
           if (error != k_ErrorType_OK)
             return S_OK;
           continue;
         }
-       
         break;
       }
       case NFileHeader::NLinkFlag::kDumpDir:
