diff --git a/CPP/7zip/Archive/Zip/ZipIn.cpp b/CPP/7zip/Archive/Zip/ZipIn.cpp
index 9e9918d..509753c 100644
--- a/CPP/7zip/Archive/Zip/ZipIn.cpp
+++ b/CPP/7zip/Archive/Zip/ZipIn.cpp
@@ -1729,6 +1729,9 @@ Hunk #2, a/CPP/7zip/Archive/Zip/ZipIn.cpp HRESULT CInArchive::FindCd(bool checkOffsetMode)
 HRESULT CInArchive::TryReadCd(CObjectVector<CItemEx> &items, const CCdInfo &cdInfo, UInt64 cdOffset, UInt64 cdSize)
 {
   items.Clear();
+  
+  // _startLocalFromCd_Disk = (UInt32)(Int32)-1;
+  // _startLocalFromCd_Offset = (UInt64)(Int64)-1;
 
   RINOK(SeekToVol(IsMultiVol ? cdInfo.CdDisk : -1, cdOffset));
 
