diff --git a/CPP/7zip/Archive/Zip/ZipIn.cpp b/CPP/7zip/Archive/Zip/ZipIn.cpp
index 9e9918d..509753c 100644
--- a/CPP/7zip/Archive/Zip/ZipIn.cpp
+++ b/CPP/7zip/Archive/Zip/ZipIn.cpp
@@ -1993,6 +1993,13 @@ Hunk #4, a/CPP/7zip/Archive/Zip/ZipIn.cpp HRESULT CVols::ParseArcName(IArchiveOpenVolumeCallback *volCallback)
     }
     else if (ext.IsEqualTo_Ascii_NoCase("exe"))
     {
+      /* possible cases:
+         - exe with zip inside
+         - sfx: a.exe, a.z02, a.z03,... , a.zip
+                a.exe is start volume.
+         - zip renamed to exe
+      */
+
       StartIsExe = true;
       BaseName = name;
       StartVolIndex = 0;
