diff --git a/CPP/7zip/Archive/Zip/ZipIn.cpp b/CPP/7zip/Archive/Zip/ZipIn.cpp
index 9e9918d..509753c 100644
--- a/CPP/7zip/Archive/Zip/ZipIn.cpp
+++ b/CPP/7zip/Archive/Zip/ZipIn.cpp
@@ -2000,7 +2000,22 @@ Hunk #5, a/CPP/7zip/Archive/Zip/ZipIn.cpp HRESULT CVols::ParseArcName(IArchiveOpenVolumeCallback *volCallback)
          We can open arc.zip, if it was requesed to open arc.exe.
          But it's possible that arc.exe and arc.zip are not parts of same archive.
          So we can disable such operation */
-      return S_FALSE; // don't open arc.zip instead of arc.exe
+
+      // 18.04: we still want to open zip renamed to exe.
+      /*
+      {
+        UString volName = name;
+        volName += IsUpperCase ? "Z01" : "z01";
+        {
+          CMyComPtr<IInStream> stream;
+          HRESULT res2 = volCallback->GetStream(volName, &stream);
+          if (res2 == S_OK)
+            DisableVolsSearch = true;
+        }
+      }
+      */
+      DisableVolsSearch = true;
+      return S_OK;
     }
     else if (ext[0] == 'z' || ext[0] == 'Z')
     {
