diff --git a/CPP/7zip/Archive/Zip/ZipIn.cpp b/CPP/7zip/Archive/Zip/ZipIn.cpp
index 9e9918d..509753c 100644
--- a/CPP/7zip/Archive/Zip/ZipIn.cpp
+++ b/CPP/7zip/Archive/Zip/ZipIn.cpp
@@ -2040,6 +2040,9 @@ Hunk #6, a/CPP/7zip/Archive/Zip/ZipIn.cpp HRESULT CVols::ParseArcName(IArchiveOpenVolumeCallback *volCallback)
 HRESULT CInArchive::ReadVols2(IArchiveOpenVolumeCallback *volCallback,
     unsigned start, int lastDisk, int zipDisk, unsigned numMissingVolsMax, unsigned &numMissingVols)
 {
+  if (Vols.DisableVolsSearch)
+    return S_OK;
+
   numMissingVols = 0;
 
   for (unsigned i = start;; i++)
