diff --git a/CPP/7zip/Archive/Zip/ZipIn.h b/CPP/7zip/Archive/Zip/ZipIn.h
index a312c36..f46f1f0 100644
--- a/CPP/7zip/Archive/Zip/ZipIn.h
+++ b/CPP/7zip/Archive/Zip/ZipIn.h
@@ -245,6 +245,9 @@ Hunk #4, a/CPP/7zip/Archive/Zip/ZipIn.h class CInArchive
   UInt64 _streamPos;
   UInt64 _cnt;
 
+  // UInt32 _startLocalFromCd_Disk;
+  // UInt64 _startLocalFromCd_Offset;
+
   size_t GetAvail() const { return _bufCached - _bufPos; }
 
   void InitBuf() { _bufPos = 0; _bufCached = 0; }
