diff --git a/CPP/7zip/Archive/Zip/ZipIn.h b/CPP/7zip/Archive/Zip/ZipIn.h
index a312c36..f46f1f0 100644
--- a/CPP/7zip/Archive/Zip/ZipIn.h
+++ b/CPP/7zip/Archive/Zip/ZipIn.h
@@ -378,6 +378,9 @@ Hunk #5, a/CPP/7zip/Archive/Zip/ZipIn.h class CInArchive
 
   UInt64 GetEmbeddedStubSize() const
   {
+    // it's possible that first item in CD doesn refers to first local item
+    // so FirstItemRelatOffset is not first local item
+
     if (ArcInfo.CdWasRead)
       return ArcInfo.FirstItemRelatOffset;
     if (IsMultiVol)
