diff --git a/CPP/7zip/Archive/Zip/ZipItem.h b/CPP/7zip/Archive/Zip/ZipItem.h
index 0cf9bd0..e576971 100644
--- a/CPP/7zip/Archive/Zip/ZipItem.h
+++ b/CPP/7zip/Archive/Zip/ZipItem.h
@@ -319,7 +319,10 @@ Hunk #6, a/CPP/7zip/Archive/Zip/ZipItem.h class CItem: public CLocalItem
   
   UINT GetCodePage() const
   {
-    Byte hostOS = GetHostOS();
+    // 18.06: now we use HostOS only from Central::MadeByVersion
+    if (!FromCentral)
+      return CP_OEMCP;
+    Byte hostOS = MadeByVersion.HostOS;
     return (UINT)((
            hostOS == NFileHeader::NHostOS::kFAT
         || hostOS == NFileHeader::NHostOS::kNTFS
