diff --git a/CPP/7zip/Archive/Zip/ZipUpdate.cpp b/CPP/7zip/Archive/Zip/ZipUpdate.cpp
index 606f782..e65c2b8 100644
--- a/CPP/7zip/Archive/Zip/ZipUpdate.cpp
+++ b/CPP/7zip/Archive/Zip/ZipUpdate.cpp
@@ -1107,8 +1107,13 @@ Hunk #8, a/CPP/7zip/Archive/Zip/ZipUpdate.cpp static HRESULT Update2(
   }
   
   RINOK(mtCompressProgressMixer.SetRatioInfo(0, NULL, NULL));
+
   archive.WriteCentralDir(items, comment);
-  return S_OK;
+  
+  complexity += kCentralHeaderSize * updateItems.Size() + 1;
+  mtProgressMixerSpec->Mixer2->SetProgressOffset(complexity);
+  return mtCompressProgressMixer.SetRatioInfo(0, NULL, NULL);
+
   #endif
 }
 
