diff --git a/CPP/7zip/Compress/DeflateDecoder.cpp b/CPP/7zip/Compress/DeflateDecoder.cpp
index 88ab1f9..fceef85 100644
--- a/CPP/7zip/Compress/DeflateDecoder.cpp
+++ b/CPP/7zip/Compress/DeflateDecoder.cpp
@@ -446,6 +446,14 @@ Hunk #1, a/CPP/7zip/Compress/DeflateDecoder.cpp void CCoder::SetOutStreamSizeResume(const UInt64 *outSize)
 
 STDMETHODIMP CCoder::SetOutStreamSize(const UInt64 *outSize)
 {
+  /*
+    18.06:
+    We want to support GetInputProcessedSize() before CCoder::Read()
+    So we call m_InBitStream.Init() even before buffer allocations
+    m_InBitStream.Init() just sets variables to default values
+    But later we will call m_InBitStream.Init() again with real buffer pointers
+  */
+  m_InBitStream.Init();
   _needInitInStream = true;
   SetOutStreamSizeResume(outSize);
   return S_OK;
