diff --git a/CPP/7zip/Compress/Lzma2Decoder.h b/CPP/7zip/Compress/Lzma2Decoder.h
index 440914e..e141488 100644
--- a/CPP/7zip/Compress/Lzma2Decoder.h
+++ b/CPP/7zip/Compress/Lzma2Decoder.h
@@ -17,28 +17,26 @@ Hunk #2, a/CPP/7zip/Compress/Lzma2Decoder.h class CDecoder:
   public ICompressSetFinishMode,
   public ICompressGetInStreamProcessedSize,
   public ICompressSetBufSize,
+
   #ifndef NO_READ_FROM_CODER
   public ICompressSetInStream,
   public ICompressSetOutStreamSize,
   public ISequentialInStream,
   #endif
+
+  #ifndef _7ZIP_ST
+  public ICompressSetCoderMt,
+  public ICompressSetMemLimit,
+  #endif
+
   public CMyUnknownImp
 {
-  Byte *_inBuf;
-  UInt32 _inPos;
-  UInt32 _inLim;
-
-  bool _finishMode;
-  bool _outSizeDefined;
-  UInt64 _outSize;
+  CLzma2DecMtHandle _dec;
   UInt64 _inProcessed;
-  UInt64 _outProcessed;
-  
-  UInt32 _outStep;
+  Byte _prop;
+  int _finishMode;
   UInt32 _inBufSize;
-  UInt32 _inBufSizeNew;
-
-  CLzma2Dec _state;
+  UInt32 _outStep;
 
 public:
   MY_QUERYINTERFACE_BEGIN2(ICompressCoder)
