diff --git a/CPP/7zip/Compress/Lzma2Decoder.h b/CPP/7zip/Compress/Lzma2Decoder.h
index 440914e..e141488 100644
--- a/CPP/7zip/Compress/Lzma2Decoder.h
+++ b/CPP/7zip/Compress/Lzma2Decoder.h
@@ -59,20 +59,28 @@ Hunk #4, a/CPP/7zip/Compress/Lzma2Decoder.h class CDecoder:
   STDMETHOD(SetDecoderProperties2)(const Byte *data, UInt32 size);
   STDMETHOD(SetFinishMode)(UInt32 finishMode);
   STDMETHOD(GetInStreamProcessedSize)(UInt64 *value);
-  STDMETHOD(SetOutStreamSize)(const UInt64 *outSize);
   STDMETHOD(SetInBufSize)(UInt32 streamIndex, UInt32 size);
   STDMETHOD(SetOutBufSize)(UInt32 streamIndex, UInt32 size);
 
-  #ifndef NO_READ_FROM_CODER
+  #ifndef _7ZIP_ST
+private:
+  int _tryMt;
+  UInt32 _numThreads;
+  UInt64 _memUsage;
+public:
+  STDMETHOD(SetNumberOfThreads)(UInt32 numThreads);
+  STDMETHOD(SetMemLimit)(UInt64 memUsage);
+  #endif
 
+  #ifndef NO_READ_FROM_CODER
 private:
   CMyComPtr<ISequentialInStream> _inStream;
+  CSeqInStreamWrap _inWrap;
 public:
-
+  STDMETHOD(SetOutStreamSize)(const UInt64 *outSize);
   STDMETHOD(SetInStream)(ISequentialInStream *inStream);
   STDMETHOD(ReleaseInStream)();
   STDMETHOD(Read)(void *data, UInt32 size, UInt32 *processedSize);
-
   #endif
 
   CDecoder();
