diff --git a/CPP/7zip/Compress/LzmaDecoder.cpp b/CPP/7zip/Compress/LzmaDecoder.cpp
index 2fbe058..83c24f1 100644
--- a/CPP/7zip/Compress/LzmaDecoder.cpp
+++ b/CPP/7zip/Compress/LzmaDecoder.cpp
@@ -30,19 +30,24 @@ Hunk #1, a/CPP/7zip/Compress/LzmaDecoder.cpp CDecoder::CDecoder():
     FinishStream(false),
     _propsWereSet(false),
     _outSizeDefined(false),
-    _outStep(1 << 22),
+    _outStep(1 << 20),
     _inBufSize(0),
     _inBufSizeNew(1 << 20)
 {
   _inProcessed = 0;
   _inPos = _inLim = 0;
 
+  /*
+  AlignOffsetAlloc_CreateVTable(&_alloc);
+  _alloc.numAlignBits = 7;
+  _alloc.offset = 0;
+  */
   LzmaDec_Construct(&_state);
 }
 
 CDecoder::~CDecoder()
 {
-  LzmaDec_Free(&_state, &g_Alloc);
+  LzmaDec_Free(&_state, &g_AlignedAlloc); // &_alloc.vt
   MyFree(_inBuf);
 }
 
