diff --git a/CPP/7zip/Compress/LzmaEncoder.cpp b/CPP/7zip/Compress/LzmaEncoder.cpp
index c64f580..e47f776 100644
--- a/CPP/7zip/Compress/LzmaEncoder.cpp
+++ b/CPP/7zip/Compress/LzmaEncoder.cpp
@@ -15,7 +15,7 @@ Hunk #1, a/CPP/7zip/Compress/LzmaEncoder.cpp namespace NLzma {
 CEncoder::CEncoder()
 {
   _encoder = NULL;
-  _encoder = LzmaEnc_Create(&g_Alloc);
+  _encoder = LzmaEnc_Create(&g_AlignedAlloc);
   if (!_encoder)
     throw 1;
 }
