diff --git a/CPP/7zip/Compress/LzmaEncoder.cpp b/CPP/7zip/Compress/LzmaEncoder.cpp
index c64f580..e47f776 100644
--- a/CPP/7zip/Compress/LzmaEncoder.cpp
+++ b/CPP/7zip/Compress/LzmaEncoder.cpp
@@ -23,7 +23,7 @@ Hunk #2, a/CPP/7zip/Compress/LzmaEncoder.cpp CEncoder::CEncoder()
 CEncoder::~CEncoder()
 {
   if (_encoder)
-    LzmaEnc_Destroy(_encoder, &g_Alloc, &g_BigAlloc);
+    LzmaEnc_Destroy(_encoder, &g_AlignedAlloc, &g_BigAlloc);
 }
 
 static inline wchar_t GetUpperChar(wchar_t c)
