diff --git a/C/LzFind.c b/C/LzFind.c
index 18ef49d..df55e86 100644
--- a/C/LzFind.c
+++ b/C/LzFind.c
@@ -648,10 +648,10 @@ Hunk #20, a/C/LzFind.c static UInt32 Bt4_MatchFinder_GetMatches(CMatchFinder *p, UInt32 *distances)
   if (offset != 0)
   {
     UPDATE_maxLen
-    distances[(size_t)offset - 2] = maxLen;
+    distances[(size_t)offset - 2] = (UInt32)maxLen;
     if (maxLen == lenLimit)
     {
-      SkipMatchesSpec(lenLimit, curMatch, MF_PARAMS(p));
+      SkipMatchesSpec((UInt32)lenLimit, curMatch, MF_PARAMS(p));
       MOVE_POS_RET;
     }
   }
