diff --git a/CPP/7zip/Compress/Rar2Decoder.cpp b/CPP/7zip/Compress/Rar2Decoder.cpp
index e906dee..fe458c8 100644
--- a/CPP/7zip/Compress/Rar2Decoder.cpp
+++ b/CPP/7zip/Compress/Rar2Decoder.cpp
@@ -130,7 +130,7 @@ Hunk #2, a/CPP/7zip/Compress/Rar2Decoder.cpp bool CDecoder::ReadTables(void)
   
   i = 0;
   
-  while (i < numLevels)
+  do
   {
     UInt32 sym = m_LevelDecoder.Decode(&m_InBitStream);
     if (sym < kTableDirectLevels)
