diff --git a/CPP/7zip/Compress/Rar2Decoder.cpp b/CPP/7zip/Compress/Rar2Decoder.cpp
index e906dee..fe458c8 100644
--- a/CPP/7zip/Compress/Rar2Decoder.cpp
+++ b/CPP/7zip/Compress/Rar2Decoder.cpp
@@ -180,6 +180,10 @@ Hunk #4, a/CPP/7zip/Compress/Rar2Decoder.cpp bool CDecoder::ReadTables(void)
       }
     }
   }
+  while (i < numLevels);
+
+  if (m_InBitStream.ExtraBitsWereRead())
+    return false;
 
   if (m_AudioMode)
     for (i = 0; i < m_NumChannels; i++)
