diff --git a/CPP/7zip/Compress/Rar2Decoder.cpp b/CPP/7zip/Compress/Rar2Decoder.cpp
index e906dee..fe458c8 100644
--- a/CPP/7zip/Compress/Rar2Decoder.cpp
+++ b/CPP/7zip/Compress/Rar2Decoder.cpp
@@ -226,24 +226,14 @@ Hunk #5, a/CPP/7zip/Compress/Rar2Decoder.cpp bool CDecoder::ReadLastTables()
   return true;
 }
 
-/*
-class CCoderReleaser
-{
-  CDecoder *m_Coder;
-public:
-  CCoderReleaser(CDecoder *coder): m_Coder(coder) {}
-  ~CCoderReleaser()
-  {
-    m_Coder->ReleaseStreams();
-  }
-};
-*/
 
 bool CDecoder::DecodeMm(UInt32 pos)
 {
   while (pos-- != 0)
   {
     UInt32 symbol = m_MMDecoders[m_MmFilter.CurrentChannel].Decode(&m_InBitStream);
+    if (m_InBitStream.ExtraBitsWereRead())
+      return false;
     if (symbol >= 256)
       return symbol == 256;
     /*
