diff --git a/CPP/7zip/Compress/Rar5Decoder.h b/CPP/7zip/Compress/Rar5Decoder.h
index c37cb0e..8174c4a 100644
--- a/CPP/7zip/Compress/Rar5Decoder.h
+++ b/CPP/7zip/Compress/Rar5Decoder.h
@@ -238,11 +238,6 @@ Hunk #3, a/CPP/7zip/Compress/Rar5Decoder.h class CDecoder:
   UInt64 _writtenFileSize;
   size_t _winSizeAllocated;
 
-  Byte _dictSizeLog;
-  bool _tableWasFilled;
-  bool _isSolid;
-  bool _wasInit;
-
   UInt32 _reps[kNumReps];
   UInt32 _lastLen;
   
