diff --git a/CPP/7zip/Compress/ShrinkDecoder.cpp b/CPP/7zip/Compress/ShrinkDecoder.cpp
index bd7c246..41f0e35 100644
--- a/CPP/7zip/Compress/ShrinkDecoder.cpp
+++ b/CPP/7zip/Compress/ShrinkDecoder.cpp
@@ -13,6 +13,7 @@ Hunk #1, a/CPP/7zip/Compress/ShrinkDecoder.cpp
 namespace NCompress {
 namespace NShrink {
 
+static const UInt32 kEmpty = 256; // kNumItems;
 static const UInt32 kBufferSize = (1 << 18);
 static const unsigned kNumMinBits = 9;
 
