diff --git a/CPP/7zip/Compress/ShrinkDecoder.cpp b/CPP/7zip/Compress/ShrinkDecoder.cpp
index bd7c246..41f0e35 100644
--- a/CPP/7zip/Compress/ShrinkDecoder.cpp
+++ b/CPP/7zip/Compress/ShrinkDecoder.cpp
@@ -67,18 +67,22 @@ Hunk #3, a/CPP/7zip/Compress/ShrinkDecoder.cpp HRESULT CDecoder::CodeReal(ISequentialInStream *inStream, ISequentialOutStream *
         break;
       }
       eofCheck = true;
-      // Is specSym(=256) allowed after end of stream
-      // Do we need to read it here
+      // Is specSym(=256) allowed after end of stream ?
+      // Do we need to read it here ?
     }
 
     if (progress)
     {
-      if (nowPos - prevPos >= (1 << 18)
-          || _inProcessed - inPrev >= (1 << 20))
+      if (nowPos - outPrev >= (1 << 20) || _inProcessed - inPrev >= (1 << 20))
       {
-        prevPos = nowPos;
+        outPrev = nowPos;
         inPrev = _inProcessed;
-        RINOK(progress->SetRatioInfo(&_inProcessed, &nowPos));
+        res = progress->SetRatioInfo(&_inProcessed, &nowPos);
+        if (res != SZ_OK)
+        {
+          // break;
+          return res;
+        }
       }
     }
 
