diff --git a/CPP/7zip/Compress/ShrinkDecoder.cpp b/CPP/7zip/Compress/ShrinkDecoder.cpp
index bd7c246..41f0e35 100644
--- a/CPP/7zip/Compress/ShrinkDecoder.cpp
+++ b/CPP/7zip/Compress/ShrinkDecoder.cpp
@@ -105,23 +105,30 @@ Hunk #4, a/CPP/7zip/Compress/ShrinkDecoder.cpp HRESULT CDecoder::CodeReal(ISequentialInStream *inStream, ISequentialOutStream *
         continue;
       }
       if (sym != 2)
-        break;
       {
+        break;
+        // continue; // info-zip just ignores such code
+      }
+      {
+        /*
+        ---------- Free leaf nodes ----------
+        Note : that code can mark _parents[lastSym] as free, and next
+        inserted node will be Orphan in that case.
+        */
+
         unsigned i;
-        for (i = 257; i < kNumItems; i++)
+        for (i = 256; i < kNumItems; i++)
           _stack[i] = 0;
         for (i = 257; i < kNumItems; i++)
         {
           unsigned par = _parents[i];
-          if (par != kNumItems)
+          if (par != kEmpty)
             _stack[par] = 1;
         }
         for (i = 257; i < kNumItems; i++)
           if (_stack[i] == 0)
-            _parents[i] = kNumItems;
-       
+            _parents[i] = kEmpty;
         head = 257;
-       
         continue;
       }
     }
