diff --git a/CPP/7zip/Compress/XpressDecoder.cpp b/CPP/7zip/Compress/XpressDecoder.cpp
index a38d941..864c6f8 100644
--- a/CPP/7zip/Compress/XpressDecoder.cpp
+++ b/CPP/7zip/Compress/XpressDecoder.cpp
@@ -108,7 +108,7 @@ Hunk #4, a/CPP/7zip/Compress/XpressDecoder.cpp HRESULT Decode(const Byte *in, size_t inSize, Byte *out, size_t outSize)
 
       BIT_STREAM_NORMALIZE
       
-      if (len > outSize - pos)
+      if (len + 3 > outSize - pos)
         return S_FALSE;
       if (dist > pos)
         return S_FALSE;
