diff --git a/C/7zArcIn.c b/C/7zArcIn.c
index e1b03d8..f74d0fa 100644
--- a/C/7zArcIn.c
+++ b/C/7zArcIn.c
@@ -666,7 +666,7 @@ Hunk #3, a/C/7zArcIn.c static SRes ReadUnpackInfo(CSzAr *p,
   MY_ALLOC(size_t, p->FoCodersOffsets, (size_t)numFolders + 1, alloc);
   MY_ALLOC(UInt32, p->FoStartPackStreamIndex, (size_t)numFolders + 1, alloc);
   MY_ALLOC(UInt32, p->FoToCoderUnpackSizes, (size_t)numFolders + 1, alloc);
-  MY_ALLOC(Byte, p->FoToMainUnpackSizeIndex, (size_t)numFolders, alloc);
+  MY_ALLOC_ZE(Byte, p->FoToMainUnpackSizeIndex, (size_t)numFolders, alloc);
   
   startBufPtr = sd.Data;
   
