diff --git a/CPP/7zip/GuiCommon.rc b/CPP/7zip/GuiCommon.rc
index 6b26ddd..b67409b 100644
--- a/CPP/7zip/GuiCommon.rc
+++ b/CPP/7zip/GuiCommon.rc
@@ -71,6 +71,9 @@ Hunk #1, a/CPP/7zip/GuiCommon.rc LANGUAGE LANG_ENGLISH, SUBLANG_ENGLISH_US
   DEFPUSHBUTTON  "OK",     IDOK,     bx2, by, bxs, bys \
   PUSHBUTTON     "Cancel", IDCANCEL, bx1, by, bxs, bys
 
+#define MY_BUTTON__CLOSE \
+  DEFPUSHBUTTON  "&Close", IDCLOSE,  bx1, by, bxs, bys
+
 
 #define MY_COMBO  CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
 #define MY_COMBO_SORTED  MY_COMBO | CBS_SORT
