diff --git a/CPP/7zip/ICoder.h b/CPP/7zip/ICoder.h
index dc9acd8..677d8cf 100644
--- a/CPP/7zip/ICoder.h
+++ b/CPP/7zip/ICoder.h
@@ -190,6 +190,12 @@ Hunk #3, a/CPP/7zip/ICoder.h CODER_INTERFACE(ICompressGetInStreamProcessedSize2, 0x27)
   STDMETHOD(GetInStreamProcessedSize2)(UInt32 streamIndex, UInt64 *value) PURE;
 };
 
+CODER_INTERFACE(ICompressSetMemLimit, 0x28)
+{
+  STDMETHOD(SetMemLimit)(UInt64 memUsage) PURE;
+};
+
+
 
 CODER_INTERFACE(ICompressGetSubStreamSize, 0x30)
 {
