diff --git a/CPP/7zip/UI/Common/ArchiveCommandLine.cpp b/CPP/7zip/UI/Common/ArchiveCommandLine.cpp
index ad21bc8..0e2a470 100644
--- a/CPP/7zip/UI/Common/ArchiveCommandLine.cpp
+++ b/CPP/7zip/UI/Common/ArchiveCommandLine.cpp
@@ -994,64 +994,6 @@ Hunk #9, a/CPP/7zip/UI/Common/ArchiveCommandLine.cpp static Int32 FindCharset(const NCommandLineParser::CParser &parser, unsigned key
   }
 }
 
-HRESULT EnumerateDirItemsAndSort(
-    NWildcard::CCensor &censor,
-    NWildcard::ECensorPathMode censorPathMode,
-    const UString &addPathPrefix,
-    UStringVector &sortedPaths,
-    UStringVector &sortedFullPaths,
-    CDirItemsStat &st,
-    IDirItemsCallback *callback)
-{
-  FStringVector paths;
-  
-  {
-    CDirItems dirItems;
-    dirItems.Callback = callback;
-    {
-      HRESULT res = EnumerateItems(censor, censorPathMode, addPathPrefix, dirItems);
-      st = dirItems.Stat;
-      RINOK(res);
-    }
-  
-    FOR_VECTOR (i, dirItems.Items)
-    {
-      const CDirItem &dirItem = dirItems.Items[i];
-      if (!dirItem.IsDir())
-        paths.Add(dirItems.GetPhyPath(i));
-    }
-  }
-  
-  if (paths.Size() == 0)
-    throw CArcCmdLineException(kCannotFindArchive);
-  
-  UStringVector fullPaths;
-  
-  unsigned i;
-  
-  for (i = 0; i < paths.Size(); i++)
-  {
-    FString fullPath;
-    NFile::NDir::MyGetFullPathName(paths[i], fullPath);
-    fullPaths.Add(fs2us(fullPath));
-  }
-  
-  CUIntVector indices;
-  SortFileNames(fullPaths, indices);
-  sortedPaths.ClearAndReserve(indices.Size());
-  sortedFullPaths.ClearAndReserve(indices.Size());
-
-  for (i = 0; i < indices.Size(); i++)
-  {
-    unsigned index = indices[i];
-    sortedPaths.AddInReserved(fs2us(paths[index]));
-    sortedFullPaths.AddInReserved(fullPaths[index]);
-    if (i > 0 && CompareFileNames(sortedFullPaths[i], sortedFullPaths[i - 1]) == 0)
-      throw CArcCmdLineException("Duplicate archive path:", sortedFullPaths[i]);
-  }
-
-  return S_OK;
-}
 
 static void SetBoolPair(NCommandLineParser::CParser &parser, unsigned switchID, CBoolPair &bp)
 {
