diff --git a/CPP/7zip/UI/Common/ArchiveCommandLine.h b/CPP/7zip/UI/Common/ArchiveCommandLine.h
index 1c96601..9ed0825 100644
--- a/CPP/7zip/UI/Common/ArchiveCommandLine.h
+++ b/CPP/7zip/UI/Common/ArchiveCommandLine.h
@@ -6,14 +6,13 @@ Hunk #1, a/CPP/7zip/UI/Common/ArchiveCommandLine.h
 #include "../../../Common/CommandLineParser.h"
 #include "../../../Common/Wildcard.h"
 
+#include "EnumDirItems.h"
+
 #include "Extract.h"
 #include "HashCalc.h"
 #include "Update.h"
 
-struct CArcCmdLineException: public UString
-{
-  CArcCmdLineException(const char *a, const wchar_t *u = NULL);
-};
+typedef CMessagePathException CArcCmdLineException;
 
 namespace NCommandType { enum EEnum
 {
