diff --git a/CPP/7zip/UI/Common/ArchiveCommandLine.h b/CPP/7zip/UI/Common/ArchiveCommandLine.h
index 1c96601..9ed0825 100644
--- a/CPP/7zip/UI/Common/ArchiveCommandLine.h
+++ b/CPP/7zip/UI/Common/ArchiveCommandLine.h
@@ -51,7 +51,7 @@ Hunk #2, a/CPP/7zip/UI/Common/ArchiveCommandLine.h struct CArcCmdLineOptions
 {
   bool HelpMode;
 
-  bool LargePages;
+  // bool LargePages;
   bool CaseSensitiveChange;
   bool CaseSensitive;
 
