diff --git a/CPP/7zip/UI/Common/ArchiveCommandLine.h b/CPP/7zip/UI/Common/ArchiveCommandLine.h
index 1c96601..9ed0825 100644
--- a/CPP/7zip/UI/Common/ArchiveCommandLine.h
+++ b/CPP/7zip/UI/Common/ArchiveCommandLine.h
@@ -110,7 +110,7 @@ Hunk #3, a/CPP/7zip/UI/Common/ArchiveCommandLine.h struct CArcCmdLineOptions
   UInt32 NumIterations;
 
   CArcCmdLineOptions():
-      LargePages(false),
+      // LargePages(false),
       CaseSensitiveChange(false),
       CaseSensitive(false),
 
