diff --git a/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp b/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp
index fa14655..de2f2e0 100644
--- a/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp
+++ b/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp
@@ -1156,7 +1156,9 @@ Hunk #6, a/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp if (askExtractMode == NArchive::NExtract::NAskMode::kExtract && !_testMode)
           bool needDelete = true;
           if (needDelete)
           {
+            if (NFind::DoesFileExist(fullProcessedPath))
             if (!DeleteFileAlways(fullProcessedPath))
+            if (GetLastError() != ERROR_FILE_NOT_FOUND)
             {
               RINOK(SendMessageError_with_LastError(kCantDeleteOutputFile, fullProcessedPath));
               return S_OK;
