diff --git a/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp b/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp
index fa14655..de2f2e0 100644
--- a/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp
+++ b/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp
@@ -1270,7 +1270,8 @@ Hunk #7, a/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp if (askExtractMode == NArchive::NExtract::NAskMode::kExtract && !_testMode)
               if (FillLinkData(data, fs2us(existPath), !isJunction))
               {
                 CReparseAttr attr;
-                if (!attr.Parse(data, data.Size()))
+                DWORD errorCode = 0;
+                if (!attr.Parse(data, data.Size(), errorCode))
                 {
                   RINOK(SendMessageError("Internal error for symbolic link file", us2fs(_item.Path)));
                   // return E_FAIL;
