diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -184,23 +184,22 @@ Hunk #4, a/CPP/7zip/UI/Common/Bench.cpp class CBenchRandomGenerator: public CBenchBuffer
 
 public:
   
-  void GenerateSimpleRandom(CBaseRandomGenerator *_RG_)
+  void GenerateSimpleRandom(UInt32 salt)
   {
-    CBaseRandomGenerator rg = *_RG_;
-    const size_t bufSize = BufferSize;
-    Byte *buf = Buffer;
+    CBaseRandomGenerator rg(salt);
+    const size_t bufSize = Size();
+    Byte *buf = (Byte *)*this;
     for (size_t i = 0; i < bufSize; i++)
       buf[i] = (Byte)rg.GetRnd();
-    *_RG_ = rg;
   }
 
-  void GenerateLz(unsigned dictBits, CBaseRandomGenerator *_RG_)
+  void GenerateLz(unsigned dictBits, UInt32 salt)
   {
-    CBaseRandomGenerator rg = *_RG_;
+    CBaseRandomGenerator rg(salt);
     UInt32 pos = 0;
     UInt32 rep0 = 1;
-    const size_t bufSize = BufferSize;
-    Byte *buf = Buffer;
+    const size_t bufSize = Size();
+    Byte *buf = (Byte *)*this;
     unsigned posBits = 1;
     
     while (pos < bufSize)
