diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -297,7 +297,7 @@ Hunk #6, a/CPP/7zip/UI/Common/Bench.cpp STDMETHODIMP CBenchmarkInStream::Read(void *data, UInt32 size, UInt32 *processed
   
 class CBenchmarkOutStream:
   public ISequentialOutStream,
-  public CBenchBuffer,
+  public CAlignedBuffer,
   public CMyUnknownImp
 {
   // bool _overflow;
