diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -522,10 +522,9 @@ Hunk #8, a/CPP/7zip/UI/Common/Bench.cpp class CBenchProgressInfo:
 {
 public:
   CBenchProgressStatus *Status;
-  HRESULT Res;
   IBenchCallback *Callback;
 
-  CBenchProgressInfo(): Callback(0) {}
+  CBenchProgressInfo(): Callback(NULL) {}
   MY_UNKNOWN_IMP
   STDMETHOD(SetRatioInfo)(const UInt64 *inSize, const UInt64 *outSize);
 };
