diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -739,26 +739,29 @@ Hunk #10, a/CPP/7zip/UI/Common/Bench.cpp struct CEncoderInfo
   const Byte *fileData;
   CBenchRandomGenerator rg;
 
-  CBenchBuffer rgCopy; // it must be 16-byte aligned !!!
+  CAlignedBuffer rgCopy; // it must be 16-byte aligned !!!
   CBenchmarkOutStream *propStreamSpec;
   CMyComPtr<ISequentialOutStream> propStream;
 
-  // for decode
+  unsigned generateDictBits;
   COneMethodInfo _method;
+  
+  // for decode
   size_t _uncompressedDataSize;
 
-  HRESULT Init(
-      const COneMethodInfo &method,
-      unsigned generateDictBits,
-      CBaseRandomGenerator *rg);
+  HRESULT Generate();
   HRESULT Encode();
   HRESULT Decode(UInt32 decoderIndex);
 
   CEncoderInfo():
+    #ifndef _7ZIP_ST
+    Common(NULL),
+    #endif
+    Salt(0),
     fileData(NULL),
     CheckCrc_Enc(true),
     CheckCrc_Dec(true),
-    outStreamSpec(0), callback(0), printCallback(0), propStreamSpec(0) {}
+    outStreamSpec(NULL), callback(NULL), printCallback(NULL), propStreamSpec(NULL) {}
 
   #ifndef _7ZIP_ST
   
