diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -773,14 +773,15 @@ Hunk #11, a/CPP/7zip/UI/Common/Bench.cpp struct CEncoderInfo
       #endif
 
       res = encoder->Encode();
-      encoder->Results[0] = res;
     }
     catch(...)
     {
       res = E_FAIL;
     }
+    encoder->Results[0] = res;
     if (res != S_OK)
       encoder->progressInfoSpec[0]->Status->SetResult(res);
+    encoder->ReadyEvent.Set();
     return 0;
   }
   
