diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -1115,12 +1115,12 @@ Hunk #19, a/CPP/7zip/UI/Common/Bench.cpp HRESULT CEncoderInfo::Decode(UInt32 decoderIndex)
     
     if (_decoderFilter)
     {
-      if (compressedSize > rgCopy.BufferSize)
+      if (compressedSize > rgCopy.Size())
         return E_FAIL;
-      memcpy(rgCopy.Buffer, outStreamSpec->Buffer, compressedSize);
+      memcpy((Byte *)rgCopy, (const Byte *)*outStreamSpec, compressedSize);
       _decoderFilter->Init();
-      My_FilterBench(_decoderFilter, rgCopy.Buffer, compressedSize);
-      RINOK(WriteStream(crcOutStream, rgCopy.Buffer, compressedSize));
+      My_FilterBench(_decoderFilter, (Byte *)rgCopy, compressedSize);
+      RINOK(WriteStream(crcOutStream, (const Byte *)rgCopy, compressedSize));
     }
     else
     {
