diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -1207,6 +1207,8 @@ Hunk #23, a/CPP/7zip/UI/Common/Bench.cpp static HRESULT MethodBench(
         numSubDecoderThreads = 2;
       }
     }
+
+  bool mtEncMode = (numEncoderThreads > 1);
   #endif
 
   CBenchEncoders encodersSpec(numEncoderThreads);
