diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -1655,7 +1655,7 @@ Hunk #33, a/CPP/7zip/UI/Common/Bench.cpp bool CrcInternalTest()
   CBaseRandomGenerator RG;
   RandGen(buf + kBufferSize0, kBufferSize1, RG);
   for (i = 0; i < kBufferSize0 + kBufferSize1 - kCheckSize; i++)
-    for (UInt32 j = 0; j < kCheckSize; j++)
+    for (unsigned j = 0; j < kCheckSize; j++)
       if (CrcCalc1(buf + i, j) != CrcCalc(buf + i, j))
         return false;
   return true;
