diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -2733,7 +2733,7 @@ Hunk #42, a/CPP/7zip/UI/Common/Bench.cpp HRESULT Bench(
     bool multiDict)
 {
   if (!CrcInternalTest())
-    return S_FALSE;
+    return E_FAIL;
 
   UInt32 numCPUs = 1;
   UInt64 ramSize = (UInt64)(sizeof(size_t)) << 29;
