diff --git a/C/LzFindMt.h b/C/LzFindMt.h
index a6645f5..ef431e3 100644
--- a/C/LzFindMt.h
+++ b/C/LzFindMt.h
@@ -19,10 +19,10 @@ Hunk #2, a/C/LzFindMt.h EXTERN_C_BEGIN
 
 typedef struct _CMtSync
 {
-  Bool wasCreated;
-  Bool needStart;
-  Bool exit;
-  Bool stopWriting;
+  BoolInt wasCreated;
+  BoolInt needStart;
+  BoolInt exit;
+  BoolInt stopWriting;
 
   CThread thread;
   CAutoResetEvent canStart;
