diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -2979,7 +2979,7 @@ Hunk #48, a/CPP/7zip/UI/Common/Bench.cpp HRESULT Bench(
         complexInCommands,
         true, numThreadsSpecified,
         method,
-        uncompressedDataSize, fileDataBuffer.Buffer,
+        uncompressedDataSize, (const Byte *)fileDataBuffer,
         kOldLzmaDictBits, printCallback, benchCallback, &benchProps);
   }
 
