diff --git a/CPP/7zip/UI/Common/Bench.cpp b/CPP/7zip/UI/Common/Bench.cpp
index d0fead7..f849a87 100644
--- a/CPP/7zip/UI/Common/Bench.cpp
+++ b/CPP/7zip/UI/Common/Bench.cpp
@@ -3383,9 +3383,9 @@ Hunk #50, a/CPP/7zip/UI/Common/Bench.cpp HRESULT Bench(
       }
 
       size_t uncompressedDataSize;
-      if (fileDataBuffer.Buffer)
+      if (fileDataBuffer.IsAllocated())
       {
-        uncompressedDataSize = fileDataBuffer.BufferSize;
+        uncompressedDataSize = fileDataBuffer.Size();
       }
       else
       {
