diff --git a/C/LzFindMt.h b/C/LzFindMt.h
index a6645f5..ef431e3 100644
--- a/C/LzFindMt.h
+++ b/C/LzFindMt.h
@@ -30,8 +30,8 @@ Hunk #3, a/C/LzFindMt.h typedef struct _CMtSync
   CAutoResetEvent wasStopped;
   CSemaphore freeSemaphore;
   CSemaphore filledSemaphore;
-  Bool csWasInitialized;
-  Bool csWasEntered;
+  BoolInt csWasInitialized;
+  BoolInt csWasEntered;
   CCriticalSection cs;
   UInt32 numProcessedBlocks;
 } CMtSync;
