diff --git a/CPP/7zip/UI/Common/EnumDirItems.cpp b/CPP/7zip/UI/Common/EnumDirItems.cpp
index 12793b7..088f077 100644
--- a/CPP/7zip/UI/Common/EnumDirItems.cpp
+++ b/CPP/7zip/UI/Common/EnumDirItems.cpp
@@ -386,15 +386,27 @@ Hunk #2, a/CPP/7zip/UI/Common/EnumDirItems.cpp HRESULT CDirItems::SetLinkInfo(CDirItem &dirItem, const NFind::CFileInfo &fi,
     return S_OK;
   const FString path = phyPrefix + fi.Name;
   CByteBuffer &buf = dirItem.ReparseData;
+  DWORD res = 0;
   if (NIO::GetReparseData(path, buf))
   {
     CReparseAttr attr;
-    if (attr.Parse(buf, buf.Size()))
+    if (attr.Parse(buf, buf.Size(), res))
       return S_OK;
+    // we ignore unknown reparse points
+    if (res != ERROR_INVALID_REPARSE_DATA)
+      res = 0;
   }
-  DWORD res = ::GetLastError();
+  else
+  {
+    res = ::GetLastError();
+    if (res == 0)
+      res = ERROR_INVALID_FUNCTION;
+  }
+
   buf.Free();
-  return AddError(path , res);
+  if (res == 0)
+    return S_OK;
+  return AddError(path, res);
 }
 
 #endif
