diff --git a/CPP/7zip/UI/Common/EnumDirItems.cpp b/CPP/7zip/UI/Common/EnumDirItems.cpp
index 12793b7..088f077 100644
--- a/CPP/7zip/UI/Common/EnumDirItems.cpp
+++ b/CPP/7zip/UI/Common/EnumDirItems.cpp
@@ -865,7 +865,8 @@ Hunk #3, a/CPP/7zip/UI/Common/EnumDirItems.cpp void CDirItems::FillFixedReparse()
       continue;
     
     CReparseAttr attr;
-    if (!attr.Parse(item.ReparseData, item.ReparseData.Size()))
+    DWORD errorCode = 0;
+    if (!attr.Parse(item.ReparseData, item.ReparseData.Size(), errorCode))
       continue;
     if (attr.IsRelative())
       continue;
