diff --git a/CPP/7zip/UI/Common/HashCalc.cpp b/CPP/7zip/UI/Common/HashCalc.cpp
index 46a69de..c340ac7 100644
--- a/CPP/7zip/UI/Common/HashCalc.cpp
+++ b/CPP/7zip/UI/Common/HashCalc.cpp
@@ -230,7 +230,7 @@ Hunk #1, a/CPP/7zip/UI/Common/HashCalc.cpp HRESULT HashCalc(
   unsigned i;
   CHashBundle hb;
   RINOK(hb.SetMethods(EXTERNAL_CODECS_LOC_VARS options.Methods));
-  hb.Init();
+  // hb.Init();
 
   hb.NumErrors = dirItems.Stat.NumErrors;
   
