diff --git a/CPP/7zip/UI/Common/HashCalc.h b/CPP/7zip/UI/Common/HashCalc.h
index 38908e2..db5b39a 100644
--- a/CPP/7zip/UI/Common/HashCalc.h
+++ b/CPP/7zip/UI/Common/HashCalc.h
@@ -77,7 +77,7 @@ Hunk #3, a/CPP/7zip/UI/Common/HashCalc.h struct CHashBundle: public IHashCalc
   virtual HRESULT GetStream(const wchar_t *name, bool isFolder) x; \
   virtual HRESULT OpenFileError(const FString &path, DWORD systemError) x; \
   virtual HRESULT SetOperationResult(UInt64 fileSize, const CHashBundle &hb, bool showHash) x; \
-  virtual HRESULT AfterLastFile(const CHashBundle &hb) x; \
+  virtual HRESULT AfterLastFile(CHashBundle &hb) x; \
 
 struct IHashCallbackUI: public IDirItemsCallback
 {
