diff --git a/CPP/7zip/UI/Common/OpenArchive.cpp b/CPP/7zip/UI/Common/OpenArchive.cpp
index 6981717..419c29e 100644
--- a/CPP/7zip/UI/Common/OpenArchive.cpp
+++ b/CPP/7zip/UI/Common/OpenArchive.cpp
@@ -563,6 +563,8 @@ Hunk #1, a/CPP/7zip/UI/Common/OpenArchive.cpp HRESULT CArc::GetItemPathToParent(UInt32 index, UInt32 parent, UStringVector &pa
     UInt32 parentType = 0;
     RINOK(GetRawProps->GetParent(curIndex, &curParent, &parentType));
 
+    // 18.06: fixed : we don't want to split name to parts
+    /*
     if (parentType != NParentType::kAltStream)
     {
       for (;;)
