diff --git a/CPP/7zip/UI/Common/OpenArchive.cpp b/CPP/7zip/UI/Common/OpenArchive.cpp
index 6981717..419c29e 100644
--- a/CPP/7zip/UI/Common/OpenArchive.cpp
+++ b/CPP/7zip/UI/Common/OpenArchive.cpp
@@ -992,7 +992,7 @@ Hunk #4, a/CPP/7zip/UI/Common/OpenArchive.cpp static void MakeCheckOrder(CCodecs *codecs,
     int index = orderIndices[i];
     if (index < 0)
       continue;
-    const CArcInfoEx &ai = codecs->Formats[index];
+    const CArcInfoEx &ai = codecs->Formats[(unsigned)index];
     if (ai.SignatureOffset != 0)
     {
       orderIndices2.Add(index);
