diff --git a/CPP/7zip/UI/Common/OpenArchive.cpp b/CPP/7zip/UI/Common/OpenArchive.cpp
index 6981717..419c29e 100644
--- a/CPP/7zip/UI/Common/OpenArchive.cpp
+++ b/CPP/7zip/UI/Common/OpenArchive.cpp
@@ -2327,7 +2327,7 @@ Hunk #7, a/CPP/7zip/UI/Common/OpenArchive.cpp HRESULT CArc::OpenStream2(const COpenOptions &op)
       if (isDifficult)
       {
         difficultFormats.Add(index);
-        difficultBools[index] = true;
+        difficultBools[(unsigned)index] = true;
       }
     }
     
