diff --git a/CPP/7zip/UI/Common/PropIDUtils.cpp b/CPP/7zip/UI/Common/PropIDUtils.cpp
index 4b796f0..7702e22 100644
--- a/CPP/7zip/UI/Common/PropIDUtils.cpp
+++ b/CPP/7zip/UI/Common/PropIDUtils.cpp
@@ -269,6 +269,14 @@ Hunk #2, a/CPP/7zip/UI/Common/PropIDUtils.cpp struct CSecID2Name
   const char *sz;
 };
 
+static int FindPairIndex(const CSecID2Name * pairs, unsigned num, UInt32 id)
+{
+  for (unsigned i = 0; i < num; i++)
+    if (pairs[i].n == id)
+      return i;
+  return -1;
+}
+
 static const CSecID2Name sid_32_Names[] =
 {
   { 544, "Administrators" },
