diff --git a/CPP/7zip/UI/Common/PropIDUtils.cpp b/CPP/7zip/UI/Common/PropIDUtils.cpp
index 4b796f0..7702e22 100644
--- a/CPP/7zip/UI/Common/PropIDUtils.cpp
+++ b/CPP/7zip/UI/Common/PropIDUtils.cpp
@@ -419,20 +419,13 @@ Hunk #4, a/CPP/7zip/UI/Common/PropIDUtils.cpp static void ParseOwner(AString &s, const Byte *p, UInt32 size, UInt32 pos)
   ParseSid(s, p + pos, size - pos, sidSize);
 }
 
-static void AddUInt32ToString(AString &s, UInt32 val)
-{
-  char sz[16];
-  ConvertUInt32ToString(val, sz);
-  s += sz;
-}
-
 static void ParseAcl(AString &s, const Byte *p, UInt32 size, const char *strName, UInt32 flags, UInt32 offset)
 {
   UInt32 control = Get16(p + 2);
   if ((flags & control) == 0)
     return;
   UInt32 pos = Get32(p + offset);
-  s += ' ';
+  s.Add_Space();
   s += strName;
   if (pos >= size)
     return;
