diff --git a/CPP/7zip/UI/Common/PropIDUtils.cpp b/CPP/7zip/UI/Common/PropIDUtils.cpp
index 4b796f0..7702e22 100644
--- a/CPP/7zip/UI/Common/PropIDUtils.cpp
+++ b/CPP/7zip/UI/Common/PropIDUtils.cpp
@@ -443,7 +443,7 @@ Hunk #5, a/CPP/7zip/UI/Common/PropIDUtils.cpp static void ParseAcl(AString &s, const Byte *p, UInt32 size, const char *strName
   if (Get16(p) != 2) // revision
     return;
   UInt32 num = Get32(p + 4);
-  AddUInt32ToString(s, num);
+  s.Add_UInt32(num);
   
   /*
   UInt32 aclSize = Get16(p + 2);
