diff --git a/CPP/7zip/UI/Common/Update.cpp b/CPP/7zip/UI/Common/Update.cpp
index ba1a81b..8c7ae45 100644
--- a/CPP/7zip/UI/Common/Update.cpp
+++ b/CPP/7zip/UI/Common/Update.cpp
@@ -288,29 +288,27 @@ Hunk #1, a/CPP/7zip/UI/Common/Update.cpp void CArchivePath::ParseFromPath(const UString &path, EArcNameMode mode)
   
   if (mode == k_ArcNameMode_Add)
     return;
-  if (mode == k_ArcNameMode_Exact)
-  {
-    BaseExtension.Empty();
-    return;
-  }
   
-  int dotPos = Name.ReverseFind_Dot();
-  if (dotPos < 0)
-    return;
-  if ((unsigned)dotPos == Name.Len() - 1)
+  if (mode != k_ArcNameMode_Exact)
   {
-    Name.DeleteBack();
-    BaseExtension.Empty();
-    return;
+    int dotPos = Name.ReverseFind_Dot();
+    if (dotPos < 0)
+      return;
+    if ((unsigned)dotPos == Name.Len() - 1)
+      Name.DeleteBack();
+    else
+    {
+      const UString ext = Name.Ptr(dotPos + 1);
+      if (BaseExtension.IsEqualTo_NoCase(ext))
+      {
+        BaseExtension = ext;
+        Name.DeleteFrom(dotPos);
+        return;
+      }
+    }
   }
-  const UString ext = Name.Ptr(dotPos + 1);
-  if (BaseExtension.IsEqualTo_NoCase(ext))
-  {
-    BaseExtension = ext;
-    Name.DeleteFrom(dotPos);
-  }
-  else
-    BaseExtension.Empty();
+
+  BaseExtension.Empty();
 }
 
 UString CArchivePath::GetFinalPath() const
