diff --git a/CPP/7zip/UI/Common/Update.cpp b/CPP/7zip/UI/Common/Update.cpp
index ba1a81b..8c7ae45 100644
--- a/CPP/7zip/UI/Common/Update.cpp
+++ b/CPP/7zip/UI/Common/Update.cpp
@@ -1189,6 +1189,16 @@ Hunk #5, a/CPP/7zip/UI/Common/Update.cpp HRESULT UpdateArchive(
         throw "there is no such archive";
       if (fi.IsDevice)
         return E_NOTIMPL;
+
+      if (!options.StdOutMode && options.UpdateArchiveItself)
+        if (fi.IsReadOnly())
+        {
+          errorInfo.SystemError = ERROR_ACCESS_DENIED;
+          errorInfo.Message = "The file is read-only";
+          errorInfo.FileNames.Add(us2fs(arcPath));
+          return errorInfo.Get_HRESULT_Error();
+        }
+
       if (options.VolumesSizes.Size() > 0)
       {
         errorInfo.FileNames.Add(us2fs(arcPath));
