diff --git a/CPP/7zip/UI/Console/Console.dsp b/CPP/7zip/UI/Console/Console.dsp
index 4145d72..34918e3 100644
--- a/CPP/7zip/UI/Console/Console.dsp
+++ b/CPP/7zip/UI/Console/Console.dsp
@@ -417,6 +417,10 @@ Hunk #1, a/CPP/7zip/UI/Console/Console.dsp SOURCE=..\..\..\Common\MyBuffer.h
 # End Source File
 # Begin Source File
 
+SOURCE=..\..\..\Common\MyBuffer2.h
+# End Source File
+# Begin Source File
+
 SOURCE=..\..\..\Common\MyCom.h
 # End Source File
 # Begin Source File
