diff --git a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
index a4bfc1c..21c2f07 100644
--- a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
@@ -60,8 +60,9 @@ Hunk #1, a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp HRESULT CExtractScanConsole::ScanError(const FString &path, DWORD systemError)
   
   if (_se)
   {
-    *_se << endl << kError << NError::MyFormatMessage(systemError) << endl <<
-        fs2us(path) << endl << endl;
+    *_se << endl << kError << NError::MyFormatMessage(systemError) << endl;
+    _se->NormalizePrint_UString(fs2us(path));
+    *_se << endl << endl;
     _se->Flush();
   }
   return HRESULT_FROM_WIN32(systemError);
