diff --git a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
index a4bfc1c..21c2f07 100644
--- a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
@@ -332,7 +332,10 @@ Hunk #5, a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp STDMETHODIMP CExtractCallbackConsole::PrepareOperation(const wchar_t *name, Int3
 
     _tempU.Empty();
     if (name)
+    {
       _tempU = name;
+      _so->Normalize_UString(_tempU);
+    }
     _so->PrintUString(_tempU, _tempA);
     if (position)
       *_so << " <" << *position << ">";
