diff --git a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
index a4bfc1c..21c2f07 100644
--- a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
@@ -453,7 +453,10 @@ Hunk #6, a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp STDMETHODIMP CExtractCallbackConsole::SetOperationResult(Int32 opRes, Int32 encr
       
       *_se << s;
       if (!_currentName.IsEmpty())
-        *_se << " : " << _currentName;
+      {
+        *_se << " : ";
+        _se->NormalizePrint_UString(_currentName);
+      }
       *_se << endl;
       _se->Flush();
     }
