diff --git a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
index a4bfc1c..21c2f07 100644
--- a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
@@ -505,7 +505,11 @@ Hunk #7, a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp HRESULT CExtractCallbackConsole::BeforeOpen(const wchar_t *name, bool testMode)
   
   ClosePercents_for_so();
   if (_so)
-    *_so << endl << (testMode ? kTesting : kExtracting) << name << endl;
+  {
+    *_so << endl << (testMode ? kTesting : kExtracting);
+    _so->NormalizePrint_wstr(name);
+    *_so << endl;
+  }
 
   if (NeedPercents())
     _percent.Command = "Open";
