diff --git a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
index a4bfc1c..21c2f07 100644
--- a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
@@ -611,7 +611,10 @@ Hunk #9, a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp HRESULT CExtractCallbackConsole::OpenResult(
       {
         *_se << endl;
         if (level != 0)
-          *_se << arc.Path << endl;
+        {
+          _se->NormalizePrint_UString(arc.Path);
+          *_se << endl;
+        }
       }
       
       if (errorFlags != 0)
