diff --git a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
index a4bfc1c..21c2f07 100644
--- a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
+++ b/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp
@@ -645,7 +645,10 @@ Hunk #10, a/CPP/7zip/UI/Console/ExtractCallbackConsole.cpp HRESULT CExtractCallbackConsole::OpenResult(
       {
         *_so << endl;
         if (level != 0)
-          *_so << arc.Path << endl;
+        {
+          _so->NormalizePrint_UString(arc.Path);
+          *_so << endl;
+        }
       }
       
       if (warningFlags != 0)
