diff --git a/CPP/7zip/UI/Console/HashCon.cpp b/CPP/7zip/UI/Console/HashCon.cpp
index 8ff9075..762b21b 100644
--- a/CPP/7zip/UI/Console/HashCon.cpp
+++ b/CPP/7zip/UI/Console/HashCon.cpp
@@ -269,7 +269,7 @@ Hunk #1, a/CPP/7zip/UI/Console/HashCon.cpp HRESULT CHashCallbackConsole::SetOperationResult(UInt64 fileSize, const CHashBun
       if (_fileName.IsEmpty())
         *_so << kEmptyFileAlias;
       else
-        *_so << _fileName;
+        _so->NormalizePrint_UString(_fileName);
     }
     *_so << endl;
   }
