diff --git a/CPP/7zip/UI/Console/List.cpp b/CPP/7zip/UI/Console/List.cpp
index 57d1d1b..416ef2c 100644
--- a/CPP/7zip/UI/Console/List.cpp
+++ b/CPP/7zip/UI/Console/List.cpp
@@ -671,9 +671,10 @@ Hunk #2, a/CPP/7zip/UI/Console/List.cpp HRESULT CFieldPrinter::PrintItemInfo(UInt32 index, const CListStat &st)
       else if (prop.vt == VT_BSTR)
       {
         TempWString.SetFromBstr(prop.bstrVal);
+        // do we need multi-line support here ?
+        g_StdOut.Normalize_UString(TempWString);
         if (techMode)
         {
-          // replace CR/LF here.
           g_StdOut.PrintUString(TempWString, TempAString);
         }
         else
