diff --git a/C/Lzma2Dec.c b/C/Lzma2Dec.c
index 63853c6..4e138a4 100644
--- a/C/Lzma2Dec.c
+++ b/C/Lzma2Dec.c
@@ -145,13 +145,12 @@ Hunk #7, a/C/Lzma2Dec.c static ELzma2State Lzma2Dec_UpdateState(CLzma2Dec *p, Byte b)
         return LZMA2_STATE_ERROR;
       lc = b % 9;
       b /= 9;
-      p->decoder.prop.pb = b / 5;
+      p->decoder.prop.pb = (Byte)(b / 5);
       lp = b % 5;
       if (lc + lp > LZMA2_LCLP_MAX)
         return LZMA2_STATE_ERROR;
-      p->decoder.prop.lc = lc;
-      p->decoder.prop.lp = lp;
-      p->needInitProp = False;
+      p->decoder.prop.lc = (Byte)lc;
+      p->decoder.prop.lp = (Byte)lp;
       return LZMA2_STATE_DATA;
     }
   }
