diff --git a/CPP/7zip/UI/Console/List.cpp b/CPP/7zip/UI/Console/List.cpp
index 57d1d1b..416ef2c 100644
--- a/CPP/7zip/UI/Console/List.cpp
+++ b/CPP/7zip/UI/Console/List.cpp
@@ -943,7 +943,8 @@ Hunk #8, a/CPP/7zip/UI/Console/List.cpp HRESULT Print_OpenArchive_Error(CStdOutStream &so, const CCodecs *codecs, const
   {
     if (arcLink.NonOpen_ErrorInfo.ErrorFormatIndex >= 0)
     {
-      so << arcLink.NonOpen_ArcPath << endl;
+      so.NormalizePrint_UString(arcLink.NonOpen_ArcPath);
+      so << endl;
       PrintArcTypeError(so, codecs->Formats[arcLink.NonOpen_ErrorInfo.ErrorFormatIndex].Name, false);
     }
     else
