diff --git a/CPP/7zip/UI/Console/Main.cpp b/CPP/7zip/UI/Console/Main.cpp
index ed47bdc..333223b 100644
--- a/CPP/7zip/UI/Console/Main.cpp
+++ b/CPP/7zip/UI/Console/Main.cpp
@@ -25,13 +25,10 @@ Hunk #2, a/CPP/7zip/UI/Console/Main.cpp
 
 #include "../../../Windows/ErrorMsg.h"
 
-#ifdef _WIN32
-#include "../../../Windows/MemoryLock.h"
-#endif
-
 #include "../../../Windows/TimeUtils.h"
 
 #include "../Common/ArchiveCommandLine.h"
+#include "../Common/Bench.h"
 #include "../Common/ExitCode.h"
 #include "../Common/Extract.h"
 
