diff --git a/CPP/7zip/UI/Console/Main.cpp b/CPP/7zip/UI/Console/Main.cpp
index ed47bdc..333223b 100644
--- a/CPP/7zip/UI/Console/Main.cpp
+++ b/CPP/7zip/UI/Console/Main.cpp
@@ -112,8 +112,7 @@ Hunk #5, a/CPP/7zip/UI/Console/Main.cpp static const char * const kHelpString =
     "  x : eXtract files with full paths\n"
     "\n"
     "<Switches>\n"
-    "  -- : Stop switches parsing\n"
-    "  @listfile : set path to listfile that contains file names\n"
+    "  -- : Stop switches and @listfile parsing\n"
     "  -ai[r[-|0]]{@listfile|!wildcard} : Include archives\n"
     "  -ax[r[-|0]]{@listfile|!wildcard} : eXclude archives\n"
     "  -ao{a|s|t|u} : set Overwrite mode\n"
