diff --git a/CPP/7zip/UI/Console/Main.cpp b/CPP/7zip/UI/Console/Main.cpp
index ed47bdc..333223b 100644
--- a/CPP/7zip/UI/Console/Main.cpp
+++ b/CPP/7zip/UI/Console/Main.cpp
@@ -243,7 +243,8 @@ Hunk #7, a/CPP/7zip/UI/Console/Main.cpp static void PrintWarningsPaths(const CErrorPathCodes &pc, CStdOutStream &so)
 {
   FOR_VECTOR(i, pc.Paths)
   {
-    so << pc.Paths[i] << " : ";
+    so.NormalizePrint_UString(fs2us(pc.Paths[i]));
+    so << " : ";
     so << NError::MyFormatMessage(pc.Codes[i]) << endl;
   }
   so << "----------------" << endl;
