diff --git a/CPP/7zip/UI/Console/Main.cpp b/CPP/7zip/UI/Console/Main.cpp
index ed47bdc..333223b 100644
--- a/CPP/7zip/UI/Console/Main.cpp
+++ b/CPP/7zip/UI/Console/Main.cpp
@@ -523,6 +523,8 @@ Hunk #9, a/CPP/7zip/UI/Console/Main.cpp int Main2(
 
   parser.Parse1(commandStrings, options);
 
+  g_StdOut.IsTerminalMode = options.IsStdOutTerminal;
+  g_StdErr.IsTerminalMode = options.IsStdErrTerminal;
 
   if (options.Number_for_Out != k_OutStream_stdout)
     g_StdStream = (options.Number_for_Out == k_OutStream_stderr ? &g_StdErr : NULL);
