diff --git a/CPP/7zip/UI/Console/Main.cpp b/CPP/7zip/UI/Console/Main.cpp
index ed47bdc..333223b 100644
--- a/CPP/7zip/UI/Console/Main.cpp
+++ b/CPP/7zip/UI/Console/Main.cpp
@@ -931,7 +931,7 @@ Hunk #11, a/CPP/7zip/UI/Console/Main.cpp int Main2(
       {
         hashCalc = &hb;
         ThrowException_if_Error(hb.SetMethods(EXTERNAL_CODECS_VARS_L options.HashMethods));
-        hb.Init();
+        // hb.Init();
       }
       
       hresultMain = Extract(
