diff --git a/CPP/7zip/UI/FileManager/PanelCrc.cpp b/CPP/7zip/UI/FileManager/PanelCrc.cpp
index 227c453..1cfbfe5 100644
--- a/CPP/7zip/UI/FileManager/PanelCrc.cpp
+++ b/CPP/7zip/UI/FileManager/PanelCrc.cpp
@@ -360,17 +360,20 @@ Hunk #7, a/CPP/7zip/UI/FileManager/PanelCrc.cpp HRESULT CApp::CalculateCrc2(const UString &methodName)
 
     t.Enumerator.EnterToDirs = !GetFlatMode();
     
-    t.ProgressDialog.ShowCompressionInfo = false;
+    t.ShowCompressionInfo = false;
     
     UString title = LangString(IDS_CHECKSUM_CALCULATING);
     
-    t.ProgressDialog.MainWindow = _window;
-    t.ProgressDialog.MainTitle = "7-Zip"; // LangString(IDS_APP_TITLE);
-    t.ProgressDialog.MainAddTitle = title;
-    t.ProgressDialog.MainAddTitle.Add_Space();
+    t.MainWindow = _window;
+    t.MainTitle = "7-Zip"; // LangString(IDS_APP_TITLE);
+    t.MainAddTitle = title;
+    t.MainAddTitle.Add_Space();
     
     RINOK(t.Create(title, _window));
+
+    t.ShowFinalResults(_window);
   }
+
   RefreshTitleAlways();
   return S_OK;
 }
