diff --git a/CPP/7zip/UI/FileManager/PanelDrag.cpp b/CPP/7zip/UI/FileManager/PanelDrag.cpp
index b6fca59..6a781e2 100644
--- a/CPP/7zip/UI/FileManager/PanelDrag.cpp
+++ b/CPP/7zip/UI/FileManager/PanelDrag.cpp
@@ -629,7 +629,7 @@ Hunk #12, a/CPP/7zip/UI/FileManager/PanelDrag.cpp static void GetNamesFromDataObject(IDataObject *dataObject, UStringVector &names
 
 bool CDropTarget::IsItSameDrive() const
 {
-  if (m_Panel == 0)
+  if (!m_Panel)
     return false;
   if (!IsFsFolderPath())
     return false;
