diff --git a/CPP/7zip/UI/GUI/ExtractGUI.cpp b/CPP/7zip/UI/GUI/ExtractGUI.cpp
index 5071d1f..1e37efb 100644
--- a/CPP/7zip/UI/GUI/ExtractGUI.cpp
+++ b/CPP/7zip/UI/GUI/ExtractGUI.cpp
@@ -52,22 +52,9 @@ Hunk #1, a/CPP/7zip/UI/GUI/ExtractGUI.cpp static void AddValuePair(UString &s, UINT resourceID, UInt64 value, bool addColo
 
 static void AddSizePair(UString &s, UINT resourceID, UInt64 value)
 {
-  {
-    wchar_t sz[32];
-    AddLangString(s, resourceID);
-    s += ": ";
-    ConvertUInt64ToString(value, sz);
-    s += MyFormatNew(IDS_FILE_SIZE, sz);
-  }
-  // s += sz;
-  if (value >= (1 << 20))
-  {
-    char sz[32];
-    ConvertUInt64ToString(value >> 20, sz);
-    s += " (";
-    s += sz;
-    s += " MB)";
-  }
+  AddLangString(s, resourceID);
+  s += ": ";
+  AddSizeValue(s, value);
   s.Add_LF();
 }
 
