diff --git a/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp b/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp
index fa14655..de2f2e0 100644
--- a/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp
+++ b/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp
@@ -1475,27 +1475,7 @@ Hunk #10, a/CPP/7zip/UI/Common/ArchiveExtractCallback.cpp STDMETHODIMP CArchiveExtractCallback::SetOperationResult(Int32 opRes)
 
   #endif
 
-  if (_outFileStream)
-  {
-    HRESULT hres = S_OK;
-    _outFileStreamSpec->SetTime(
-        (WriteCTime && _fi.CTimeDefined) ? &_fi.CTime : NULL,
-        (WriteATime && _fi.ATimeDefined) ? &_fi.ATime : NULL,
-        (WriteMTime && _fi.MTimeDefined) ? &_fi.MTime : (_arc->MTimeDefined ? &_arc->MTime : NULL));
-    const UInt64 processedSize = _outFileStreamSpec->ProcessedSize;
-    if (_fileLengthWasSet && _curSize > processedSize)
-    {
-      bool res = _outFileStreamSpec->File.SetLength(processedSize);
-      _fileLengthWasSet = res;
-      if (!res)
-        hres = SendMessageError_with_LastError(kCantSetFileLen, us2fs(_item.Path));
-    }
-    _curSize = processedSize;
-    _curSizeDefined = true;
-    RINOK(_outFileStreamSpec->Close());
-    _outFileStream.Release();
-    RINOK(hres);
-  }
+  RINOK(CloseFile());
   
   #ifdef _USE_SECURITY_CODE
   if (!_stdOutMode && _extractMode && _ntOptions.NtSecurity.Val && _arc->GetRawProps)
