diff --git a/CPP/7zip/UI/Common/Bench.h b/CPP/7zip/UI/Common/Bench.h
index ec51fae..18a40a8 100644
--- a/CPP/7zip/UI/Common/Bench.h
+++ b/CPP/7zip/UI/Common/Bench.h
@@ -68,5 +68,10 @@ Hunk #1, a/CPP/7zip/UI/Common/Bench.h void GetSysInfo(AString &s1, AString &s2);
 void GetCpuName(AString &s);
 void GetCpuFeatures(AString &s);
 
+#ifdef _7ZIP_LARGE_PAGES
+void Add_LargePages_String(AString &s);
+#else
+// #define Add_LargePages_String
+#endif
 
 #endif
