diff --git a/CPP/7zip/UI/Common/Update.cpp b/CPP/7zip/UI/Common/Update.cpp
index ba1a81b..8c7ae45 100644
--- a/CPP/7zip/UI/Common/Update.cpp
+++ b/CPP/7zip/UI/Common/Update.cpp
@@ -1568,26 +1568,39 @@ Hunk #9, a/CPP/7zip/UI/Common/Update.cpp HRESULT UpdateArchive(
     }
 
     CCurrentDirRestorer curDirRestorer;
+
+    AStringVector paths;
+    AStringVector names;
     
     for (i = 0; i < fullPaths.Size(); i++)
     {
       const UString arcPath2 = fs2us(fullPaths[i]);
       const UString fileName = ExtractFileNameFromPath(arcPath2);
-      const AString path (GetAnsiString(arcPath2));
-      const AString name (GetAnsiString(fileName));
+      paths.Add(GetAnsiString(arcPath2));
+      names.Add(GetAnsiString(fileName));
+      // const AString path (GetAnsiString(arcPath2));
+      // const AString name (GetAnsiString(fileName));
       // Warning!!! MAPISendDocuments function changes Current directory
       // fnSend(0, ";", (LPSTR)(LPCSTR)path, (LPSTR)(LPCSTR)name, 0);
+    }
 
-      MapiFileDesc f;
+    CRecordVector<MapiFileDesc> files;
+    files.ClearAndSetSize(paths.Size());
+    
+    for (i = 0; i < paths.Size(); i++)
+    {
+      MapiFileDesc &f = files[i];
       memset(&f, 0, sizeof(f));
       f.nPosition = 0xFFFFFFFF;
-      f.lpszPathName = (char *)(const char *)path;
-      f.lpszFileName = (char *)(const char *)name;
-      
+      f.lpszPathName = (char *)(const char *)paths[i];
+      f.lpszFileName = (char *)(const char *)names[i];
+    }
+
+    {
       MapiMessage m;
       memset(&m, 0, sizeof(m));
-      m.nFileCount = 1;
-      m.lpFiles = &f;
+      m.nFileCount = files.Size();
+      m.lpFiles = &files.Front();
       
       const AString addr (GetAnsiString(options.EMailAddress));
       MapiRecipDesc rec;
